// testit-api-client/dist/model/backgroundJobType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackgroundJobType = void 0;
var BackgroundJobType;
(function (BackgroundJobType) {
    BackgroundJobType[BackgroundJobType["ExportXlsxTestResultsByTestPlan"] = 'ExportXlsxTestResultsByTestPlan'] = "ExportXlsxTestResultsByTestPlan";
    BackgroundJobType[BackgroundJobType["ExportXlsxWorkItemsByProject"] = 'ExportXlsxWorkItemsByProject'] = "ExportXlsxWorkItemsByProject";
    BackgroundJobType[BackgroundJobType["ExportXlsxTestPointsByTestPlan"] = 'ExportXlsxTestPointsByTestPlan'] = "ExportXlsxTestPointsByTestPlan";
    BackgroundJobType[BackgroundJobType["ExportJsonProject"] = 'ExportJsonProject'] = "ExportJsonProject";
    BackgroundJobType[BackgroundJobType["ExportZipProject"] = 'ExportZipProject'] = "ExportZipProject";
    BackgroundJobType[BackgroundJobType["ExportJsonProjectWithTestPlans"] = 'ExportJsonProjectWithTestPlans'] = "ExportJsonProjectWithTestPlans";
    BackgroundJobType[BackgroundJobType["ExportZipProjectWithTestPlans"] = 'ExportZipProjectWithTestPlans'] = "ExportZipProjectWithTestPlans";
    BackgroundJobType[BackgroundJobType["ImportJsonProject"] = 'ImportJsonProject'] = "ImportJsonProject";
    BackgroundJobType[BackgroundJobType["ImportZipProject"] = 'ImportZipProject'] = "ImportZipProject";
    BackgroundJobType[BackgroundJobType["ImportXlsxProject"] = 'ImportXlsxProject'] = "ImportXlsxProject";
    BackgroundJobType[BackgroundJobType["ImportTestRailXmlProject"] = 'ImportTestRailXmlProject'] = "ImportTestRailXmlProject";
    BackgroundJobType[BackgroundJobType["PurgeProject"] = 'PurgeProject'] = "PurgeProject";
    BackgroundJobType[BackgroundJobType["ExportProjects"] = 'ExportProjects'] = "ExportProjects";
    BackgroundJobType[BackgroundJobType["ImportProjects"] = 'ImportProjects'] = "ImportProjects";
})(BackgroundJobType = exports.BackgroundJobType || (exports.BackgroundJobType = {}));
