// testit-api-client/dist/model/models.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VoidAuth = exports.OAuth = exports.ApiKeyAuth = exports.HttpBearerAuth = exports.HttpBasicAuth = exports.ObjectSerializer = void 0;
__exportStar(require("./actionUpdate"), exports);
__exportStar(require("./apiV2AttachmentsPostRequest"), exports);
__exportStar(require("./attachmentChangeViewModel"), exports);
__exportStar(require("./attachmentChangeViewModelArrayChangedFieldViewModel"), exports);
__exportStar(require("./attachmentModel"), exports);
__exportStar(require("./attachmentModelAutoTestStepResultsModel"), exports);
__exportStar(require("./attachmentPutModel"), exports);
__exportStar(require("./attachmentPutModelAutoTestStepResultsModel"), exports);
__exportStar(require("./autoTestAverageDurationModel"), exports);
__exportStar(require("./autoTestChangeViewModel"), exports);
__exportStar(require("./autoTestChangeViewModelArrayChangedFieldViewModel"), exports);
__exportStar(require("./autoTestIdModel"), exports);
__exportStar(require("./autoTestModel"), exports);
__exportStar(require("./autoTestModelV2GetModel"), exports);
__exportStar(require("./autoTestNamespaceModel"), exports);
__exportStar(require("./autoTestPostModel"), exports);
__exportStar(require("./autoTestPutModel"), exports);
__exportStar(require("./autoTestRelatedToTestResult"), exports);
__exportStar(require("./autoTestResultsForTestRunModel"), exports);
__exportStar(require("./autoTestShortModel"), exports);
__exportStar(require("./autoTestStepModel"), exports);
__exportStar(require("./autotestFilterModel"), exports);
__exportStar(require("./autotestFilterModelCreatedDate"), exports);
__exportStar(require("./autotestFilterModelModifiedDate"), exports);
__exportStar(require("./autotestFilterModelStabilityPercentage"), exports);
__exportStar(require("./autotestHistoricalResultSelectModel"), exports);
__exportStar(require("./autotestResultHistoricalGetModel"), exports);
__exportStar(require("./autotestResultOutcome"), exports);
__exportStar(require("./autotestResultReasonSubGetModel"), exports);
__exportStar(require("./autotestSelectModel"), exports);
__exportStar(require("./autotestsExtractionModel"), exports);
__exportStar(require("./autotestsExtractionModelIds"), exports);
__exportStar(require("./autotestsSelectModel"), exports);
__exportStar(require("./availableTestResultOutcome"), exports);
__exportStar(require("./backgroundJobAttachmentModel"), exports);
__exportStar(require("./backgroundJobFilterModel"), exports);
__exportStar(require("./backgroundJobGetModel"), exports);
__exportStar(require("./backgroundJobState"), exports);
__exportStar(require("./backgroundJobType"), exports);
__exportStar(require("./booleanChangedFieldViewModel"), exports);
__exportStar(require("./booleanNullableChangedFieldViewModel"), exports);
__exportStar(require("./configurationByParametersModel"), exports);
__exportStar(require("./configurationExtractionModel"), exports);
__exportStar(require("./configurationExtractionModelIds"), exports);
__exportStar(require("./configurationExtractionModelProjectIds"), exports);
__exportStar(require("./configurationFilterModel"), exports);
__exportStar(require("./configurationModel"), exports);
__exportStar(require("./configurationPostModel"), exports);
__exportStar(require("./configurationPutModel"), exports);
__exportStar(require("./configurationSelectModel"), exports);
__exportStar(require("./configurationSelectModelExtractionModel"), exports);
__exportStar(require("./configurationSelectModelFilter"), exports);
__exportStar(require("./customAttributeChangeModel"), exports);
__exportStar(require("./customAttributeGetModel"), exports);
__exportStar(require("./customAttributeModel"), exports);
__exportStar(require("./customAttributeOptionModel"), exports);
__exportStar(require("./customAttributeOptionPostModel"), exports);
__exportStar(require("./customAttributePostModel"), exports);
__exportStar(require("./customAttributePutModel"), exports);
__exportStar(require("./customAttributeSearchQueryModel"), exports);
__exportStar(require("./customAttributeTemplateModel"), exports);
__exportStar(require("./customAttributeTemplatePostModel"), exports);
__exportStar(require("./customAttributeTemplatePutModel"), exports);
__exportStar(require("./customAttributeTemplateSearchQueryModel"), exports);
__exportStar(require("./customAttributeTestPlanProjectRelationPutModel"), exports);
__exportStar(require("./customAttributeTypesEnum"), exports);
__exportStar(require("./dateTimeRangeSelectorModel"), exports);
__exportStar(require("./deletionState"), exports);
__exportStar(require("./externalLinkModel"), exports);
__exportStar(require("./failureCategoryModel"), exports);
__exportStar(require("./failureClassModel"), exports);
__exportStar(require("./failureClassRegexModel"), exports);
__exportStar(require("./filterModel"), exports);
__exportStar(require("./flakyBulkModel"), exports);
__exportStar(require("./getXlsxTestPointsByTestPlanModel"), exports);
__exportStar(require("./globalCustomAttributePostModel"), exports);
__exportStar(require("./globalCustomAttributeUpdateModel"), exports);
__exportStar(require("./globalSearchItemResult"), exports);
__exportStar(require("./globalSearchRequest"), exports);
__exportStar(require("./globalSearchResponse"), exports);
__exportStar(require("./guidChangedFieldViewModel"), exports);
__exportStar(require("./guidExtractionModel"), exports);
__exportStar(require("./imageResizeType"), exports);
__exportStar(require("./int32ChangedFieldViewModel"), exports);
__exportStar(require("./int32RangeSelectorModel"), exports);
__exportStar(require("./int64ChangedFieldViewModel"), exports);
__exportStar(require("./int64RangeSelectorModel"), exports);
__exportStar(require("./iterationModel"), exports);
__exportStar(require("./iterationPutModel"), exports);
__exportStar(require("./labelPostModel"), exports);
__exportStar(require("./labelShortModel"), exports);
__exportStar(require("./lastTestResultModel"), exports);
__exportStar(require("./linkModel"), exports);
__exportStar(require("./linkPostModel"), exports);
__exportStar(require("./linkPutModel"), exports);
__exportStar(require("./linkShortModel"), exports);
__exportStar(require("./linkSubGetModel"), exports);
__exportStar(require("./linkType"), exports);
__exportStar(require("./namedEntityModel"), exports);
__exportStar(require("./noContentResult"), exports);
__exportStar(require("./notificationModel"), exports);
__exportStar(require("./notificationQueryFilterModel"), exports);
__exportStar(require("./notificationTypeModel"), exports);
__exportStar(require("./operation"), exports);
__exportStar(require("./parameterFilterModel"), exports);
__exportStar(require("./parameterGroupModel"), exports);
__exportStar(require("./parameterIterationModel"), exports);
__exportStar(require("./parameterModel"), exports);
__exportStar(require("./parameterPostModel"), exports);
__exportStar(require("./parameterPutModel"), exports);
__exportStar(require("./parameterShortModel"), exports);
__exportStar(require("./periodViewModel"), exports);
__exportStar(require("./periodViewModelChangedFieldViewModel"), exports);
__exportStar(require("./problemDetails"), exports);
__exportStar(require("./projectAttributesFilterModel"), exports);
__exportStar(require("./projectCustomAttributeTemplateGetModel"), exports);
__exportStar(require("./projectCustomAttributesTemplatesFilterModel"), exports);
__exportStar(require("./projectExportQueryModel"), exports);
__exportStar(require("./projectExportWithTestPlansPostModel"), exports);
__exportStar(require("./projectExtractionModel"), exports);
__exportStar(require("./projectModel"), exports);
__exportStar(require("./projectPostModel"), exports);
__exportStar(require("./projectPutModel"), exports);
__exportStar(require("./projectSelectModel"), exports);
__exportStar(require("./projectShortestModel"), exports);
__exportStar(require("./projectTestPlansFilterModel"), exports);
__exportStar(require("./projectsFilterModel"), exports);
__exportStar(require("./projectsFilterModelAutotestsCount"), exports);
__exportStar(require("./projectsFilterModelChecklistsCount"), exports);
__exportStar(require("./projectsFilterModelCreatedDate"), exports);
__exportStar(require("./projectsFilterModelSharedStepsCount"), exports);
__exportStar(require("./projectsFilterModelTestCasesCount"), exports);
__exportStar(require("./publicTestPointModel"), exports);
__exportStar(require("./publicTestRunModel"), exports);
__exportStar(require("./requestTypeModel"), exports);
__exportStar(require("./searchAutoTestsQueryIncludesModel"), exports);
__exportStar(require("./searchCustomAttributeTemplateGetModel"), exports);
__exportStar(require("./searchWebhooksQueryModel"), exports);
__exportStar(require("./sectionModel"), exports);
__exportStar(require("./sectionMoveModel"), exports);
__exportStar(require("./sectionPostModel"), exports);
__exportStar(require("./sectionPutModel"), exports);
__exportStar(require("./sectionRenameModel"), exports);
__exportStar(require("./sectionSharedStep"), exports);
__exportStar(require("./sectionWithStepsModel"), exports);
__exportStar(require("./sharedStepChangeViewModel"), exports);
__exportStar(require("./sharedStepModel"), exports);
__exportStar(require("./sharedStepReferenceModel"), exports);
__exportStar(require("./sharedStepReferenceSectionModel"), exports);
__exportStar(require("./sharedStepReferenceSectionsQueryFilterModel"), exports);
__exportStar(require("./sharedStepReferenceSectionsQueryFilterModelCreatedDate"), exports);
__exportStar(require("./sharedStepReferenceSectionsQueryFilterModelModifiedDate"), exports);
__exportStar(require("./sharedStepReferencesQueryFilterModel"), exports);
__exportStar(require("./sharedStepResultModel"), exports);
__exportStar(require("./shortConfiguration"), exports);
__exportStar(require("./stepCommentModel"), exports);
__exportStar(require("./stepModel"), exports);
__exportStar(require("./stepPostModel"), exports);
__exportStar(require("./stepPutModel"), exports);
__exportStar(require("./stepResultModel"), exports);
__exportStar(require("./stringArrayChangedFieldViewModel"), exports);
__exportStar(require("./stringChangedFieldViewModel"), exports);
__exportStar(require("./stringChangedFieldWithDiffsViewModel"), exports);
__exportStar(require("./tagExtractionModel"), exports);
__exportStar(require("./tagModel"), exports);
__exportStar(require("./tagPostModel"), exports);
__exportStar(require("./tagPutModel"), exports);
__exportStar(require("./tagSelectModel"), exports);
__exportStar(require("./tagShortModel"), exports);
__exportStar(require("./tagsFilterModel"), exports);
__exportStar(require("./tagsFilterModelCreatedDate"), exports);
__exportStar(require("./testPlanChangeModel"), exports);
__exportStar(require("./testPlanChangedFieldsViewModel"), exports);
__exportStar(require("./testPlanExtractionModel"), exports);
__exportStar(require("./testPlanGroupByStatus"), exports);
__exportStar(require("./testPlanGroupByTestSuite"), exports);
__exportStar(require("./testPlanGroupByTester"), exports);
__exportStar(require("./testPlanGroupByTesterAndStatus"), exports);
__exportStar(require("./testPlanLink"), exports);
__exportStar(require("./testPlanModel"), exports);
__exportStar(require("./testPlanPostModel"), exports);
__exportStar(require("./testPlanPutModel"), exports);
__exportStar(require("./testPlanSelectModel"), exports);
__exportStar(require("./testPlanShortModel"), exports);
__exportStar(require("./testPlanStatusModel"), exports);
__exportStar(require("./testPlanWithAnalyticModel"), exports);
__exportStar(require("./testPlanWithTestSuiteTreeModel"), exports);
__exportStar(require("./testPointAnalyticResult"), exports);
__exportStar(require("./testPointByTestSuiteModel"), exports);
__exportStar(require("./testPointChangeViewModel"), exports);
__exportStar(require("./testPointChangeViewModelChangedFieldViewModel"), exports);
__exportStar(require("./testPointFilterModel"), exports);
__exportStar(require("./testPointFilterModelCreatedDate"), exports);
__exportStar(require("./testPointFilterModelDuration"), exports);
__exportStar(require("./testPointFilterModelModifiedDate"), exports);
__exportStar(require("./testPointFilterModelWorkItemCreatedDate"), exports);
__exportStar(require("./testPointFilterModelWorkItemMedianDuration"), exports);
__exportStar(require("./testPointFilterModelWorkItemModifiedDate"), exports);
__exportStar(require("./testPointPutModel"), exports);
__exportStar(require("./testPointRelatedToTestResult"), exports);
__exportStar(require("./testPointResultModel"), exports);
__exportStar(require("./testPointSelectModel"), exports);
__exportStar(require("./testPointSelector"), exports);
__exportStar(require("./testPointShortGetModel"), exports);
__exportStar(require("./testPointShortModel"), exports);
__exportStar(require("./testPointStatus"), exports);
__exportStar(require("./testPointWithLastResultModel"), exports);
__exportStar(require("./testPointsExtractionModel"), exports);
__exportStar(require("./testPointsExtractionModelIds"), exports);
__exportStar(require("./testResultChangeViewModel"), exports);
__exportStar(require("./testResultChangeViewModelChangedFieldViewModel"), exports);
__exportStar(require("./testResultChronologyModel"), exports);
__exportStar(require("./testResultConfiguration"), exports);
__exportStar(require("./testResultHistoryReportModel"), exports);
__exportStar(require("./testResultModel"), exports);
__exportStar(require("./testResultOutcome"), exports);
__exportStar(require("./testResultShortGetModel"), exports);
__exportStar(require("./testResultShortModel"), exports);
__exportStar(require("./testResultStepCommentPutModel"), exports);
__exportStar(require("./testResultUpdateModel"), exports);
__exportStar(require("./testResultV2GetModel"), exports);
__exportStar(require("./testResultV2ShortModel"), exports);
__exportStar(require("./testResultsFilterModel"), exports);
__exportStar(require("./testResultsFilterModelCompletedOn"), exports);
__exportStar(require("./testResultsFilterModelCreatedDate"), exports);
__exportStar(require("./testResultsFilterModelDuration"), exports);
__exportStar(require("./testResultsFilterModelModifiedDate"), exports);
__exportStar(require("./testResultsFilterModelStartedOn"), exports);
__exportStar(require("./testResultsLocalFilterModel"), exports);
__exportStar(require("./testResultsStatisticsGetModel"), exports);
__exportStar(require("./testResultsStatisticsGetModelFailureCategories"), exports);
__exportStar(require("./testResultsStatisticsGetModelStatuses"), exports);
__exportStar(require("./testRunAnalyticResultModel"), exports);
__exportStar(require("./testRunExtractionModel"), exports);
__exportStar(require("./testRunExtractionModelIds"), exports);
__exportStar(require("./testRunFillByAutoTestsPostModel"), exports);
__exportStar(require("./testRunFillByConfigurationsPostModel"), exports);
__exportStar(require("./testRunFillByWorkItemsPostModel"), exports);
__exportStar(require("./testRunFilterModel"), exports);
__exportStar(require("./testRunFilterModelAutoTestsCount"), exports);
__exportStar(require("./testRunFilterModelCompletedDate"), exports);
__exportStar(require("./testRunFilterModelCreatedDate"), exports);
__exportStar(require("./testRunFilterModelStartedDate"), exports);
__exportStar(require("./testRunGroupByFailureClassModel"), exports);
__exportStar(require("./testRunGroupByStatusModel"), exports);
__exportStar(require("./testRunModel"), exports);
__exportStar(require("./testRunSearchQueryModel"), exports);
__exportStar(require("./testRunSelectModel"), exports);
__exportStar(require("./testRunShortGetModel"), exports);
__exportStar(require("./testRunShortModel"), exports);
__exportStar(require("./testRunState"), exports);
__exportStar(require("./testRunStatisticsErrorCategoriesGetModel"), exports);
__exportStar(require("./testRunStatisticsStatusesGetModel"), exports);
__exportStar(require("./testRunTestResultsPartialBulkSetModel"), exports);
__exportStar(require("./testRunTestResultsPartialBulkSetModelSelector"), exports);
__exportStar(require("./testRunTestResultsSelectModel"), exports);
__exportStar(require("./testRunTestResultsSelectModelFilter"), exports);
__exportStar(require("./testRunTestResultsSelectModelTestResultIdsExtractionModel"), exports);
__exportStar(require("./testRunUpdateMultipleModel"), exports);
__exportStar(require("./testRunV2GetModel"), exports);
__exportStar(require("./testRunV2PostShortModel"), exports);
__exportStar(require("./testRunV2PutModel"), exports);
__exportStar(require("./testSuiteChangeViewModel"), exports);
__exportStar(require("./testSuiteChangeViewModelChangedFieldViewModel"), exports);
__exportStar(require("./testSuiteType"), exports);
__exportStar(require("./testSuiteV2GetModel"), exports);
__exportStar(require("./testSuiteV2PostModel"), exports);
__exportStar(require("./testSuiteV2PutModel"), exports);
__exportStar(require("./testSuiteV2TreeModel"), exports);
__exportStar(require("./testSuiteWithChildrenModel"), exports);
__exportStar(require("./testSuiteWorkItemsSearchModel"), exports);
__exportStar(require("./testSuiteWorkItemsSearchModelDuration"), exports);
__exportStar(require("./testSuiteWorkItemsSearchModelMedianDuration"), exports);
__exportStar(require("./updateAttachmentShortModel"), exports);
__exportStar(require("./updateLinkShortModel"), exports);
__exportStar(require("./userRankModel"), exports);
__exportStar(require("./userWithRankModel"), exports);
__exportStar(require("./validateAntiForgeryTokenAttribute"), exports);
__exportStar(require("./validationProblemDetails"), exports);
__exportStar(require("./webHookEventType"), exports);
__exportStar(require("./webHookEventTypeModel"), exports);
__exportStar(require("./webHookLogModel"), exports);
__exportStar(require("./webHookModel"), exports);
__exportStar(require("./webHookPostModel"), exports);
__exportStar(require("./webHookTestModel"), exports);
__exportStar(require("./webhookResponse"), exports);
__exportStar(require("./webhookVariablesType"), exports);
__exportStar(require("./workItemChangeModel"), exports);
__exportStar(require("./workItemChangedAttributeViewModel"), exports);
__exportStar(require("./workItemChangedFieldsViewModel"), exports);
__exportStar(require("./workItemCommentModel"), exports);
__exportStar(require("./workItemCommentPostModel"), exports);
__exportStar(require("./workItemCommentPutModel"), exports);
__exportStar(require("./workItemEntityTypes"), exports);
__exportStar(require("./workItemExtractionModel"), exports);
__exportStar(require("./workItemExtractionModelIds"), exports);
__exportStar(require("./workItemExtractionModelSectionIds"), exports);
__exportStar(require("./workItemFilterModel"), exports);
__exportStar(require("./workItemGroupGetModel"), exports);
__exportStar(require("./workItemGroupGetModelSelectModel"), exports);
__exportStar(require("./workItemGroupModel"), exports);
__exportStar(require("./workItemGroupType"), exports);
__exportStar(require("./workItemIdModel"), exports);
__exportStar(require("./workItemIdentifierModel"), exports);
__exportStar(require("./workItemLikeModel"), exports);
__exportStar(require("./workItemLinkChangeViewModel"), exports);
__exportStar(require("./workItemLinkChangeViewModelArrayChangedFieldViewModel"), exports);
__exportStar(require("./workItemLocalFilterModel"), exports);
__exportStar(require("./workItemLocalSelectModel"), exports);
__exportStar(require("./workItemLocalSelectModelExtractionModel"), exports);
__exportStar(require("./workItemLocalSelectModelFilter"), exports);
__exportStar(require("./workItemModel"), exports);
__exportStar(require("./workItemMovePostModel"), exports);
__exportStar(require("./workItemPostModel"), exports);
__exportStar(require("./workItemPriorityModel"), exports);
__exportStar(require("./workItemPutModel"), exports);
__exportStar(require("./workItemSearchQueryModel"), exports);
__exportStar(require("./workItemSelectModel"), exports);
__exportStar(require("./workItemShortModel"), exports);
__exportStar(require("./workItemState"), exports);
__exportStar(require("./workItemStates"), exports);
__exportStar(require("./workItemStepChangeViewModel"), exports);
__exportStar(require("./workItemStepChangeViewModelArrayChangedFieldWithDiffsViewModel"), exports);
__exportStar(require("./workItemVersionModel"), exports);
const actionUpdate_1 = require("./actionUpdate");
const apiV2AttachmentsPostRequest_1 = require("./apiV2AttachmentsPostRequest");
const attachmentChangeViewModel_1 = require("./attachmentChangeViewModel");
const attachmentChangeViewModelArrayChangedFieldViewModel_1 = require("./attachmentChangeViewModelArrayChangedFieldViewModel");
const attachmentModel_1 = require("./attachmentModel");
const attachmentModelAutoTestStepResultsModel_1 = require("./attachmentModelAutoTestStepResultsModel");
const attachmentPutModel_1 = require("./attachmentPutModel");
const attachmentPutModelAutoTestStepResultsModel_1 = require("./attachmentPutModelAutoTestStepResultsModel");
const autoTestAverageDurationModel_1 = require("./autoTestAverageDurationModel");
const autoTestChangeViewModel_1 = require("./autoTestChangeViewModel");
const autoTestChangeViewModelArrayChangedFieldViewModel_1 = require("./autoTestChangeViewModelArrayChangedFieldViewModel");
const autoTestIdModel_1 = require("./autoTestIdModel");
const autoTestModel_1 = require("./autoTestModel");
const autoTestModelV2GetModel_1 = require("./autoTestModelV2GetModel");
const autoTestNamespaceModel_1 = require("./autoTestNamespaceModel");
const autoTestPostModel_1 = require("./autoTestPostModel");
const autoTestPutModel_1 = require("./autoTestPutModel");
const autoTestRelatedToTestResult_1 = require("./autoTestRelatedToTestResult");
const autoTestResultsForTestRunModel_1 = require("./autoTestResultsForTestRunModel");
const autoTestShortModel_1 = require("./autoTestShortModel");
const autoTestStepModel_1 = require("./autoTestStepModel");
const autotestFilterModel_1 = require("./autotestFilterModel");
const autotestFilterModelCreatedDate_1 = require("./autotestFilterModelCreatedDate");
const autotestFilterModelModifiedDate_1 = require("./autotestFilterModelModifiedDate");
const autotestFilterModelStabilityPercentage_1 = require("./autotestFilterModelStabilityPercentage");
const autotestHistoricalResultSelectModel_1 = require("./autotestHistoricalResultSelectModel");
const autotestResultHistoricalGetModel_1 = require("./autotestResultHistoricalGetModel");
const autotestResultOutcome_1 = require("./autotestResultOutcome");
const autotestResultReasonSubGetModel_1 = require("./autotestResultReasonSubGetModel");
const autotestSelectModel_1 = require("./autotestSelectModel");
const autotestsExtractionModel_1 = require("./autotestsExtractionModel");
const autotestsExtractionModelIds_1 = require("./autotestsExtractionModelIds");
const autotestsSelectModel_1 = require("./autotestsSelectModel");
const availableTestResultOutcome_1 = require("./availableTestResultOutcome");
const backgroundJobAttachmentModel_1 = require("./backgroundJobAttachmentModel");
const backgroundJobFilterModel_1 = require("./backgroundJobFilterModel");
const backgroundJobGetModel_1 = require("./backgroundJobGetModel");
const backgroundJobState_1 = require("./backgroundJobState");
const backgroundJobType_1 = require("./backgroundJobType");
const booleanChangedFieldViewModel_1 = require("./booleanChangedFieldViewModel");
const booleanNullableChangedFieldViewModel_1 = require("./booleanNullableChangedFieldViewModel");
const configurationByParametersModel_1 = require("./configurationByParametersModel");
const configurationExtractionModel_1 = require("./configurationExtractionModel");
const configurationExtractionModelIds_1 = require("./configurationExtractionModelIds");
const configurationExtractionModelProjectIds_1 = require("./configurationExtractionModelProjectIds");
const configurationFilterModel_1 = require("./configurationFilterModel");
const configurationModel_1 = require("./configurationModel");
const configurationPostModel_1 = require("./configurationPostModel");
const configurationPutModel_1 = require("./configurationPutModel");
const configurationSelectModel_1 = require("./configurationSelectModel");
const configurationSelectModelExtractionModel_1 = require("./configurationSelectModelExtractionModel");
const configurationSelectModelFilter_1 = require("./configurationSelectModelFilter");
const customAttributeChangeModel_1 = require("./customAttributeChangeModel");
const customAttributeGetModel_1 = require("./customAttributeGetModel");
const customAttributeModel_1 = require("./customAttributeModel");
const customAttributeOptionModel_1 = require("./customAttributeOptionModel");
const customAttributeOptionPostModel_1 = require("./customAttributeOptionPostModel");
const customAttributePostModel_1 = require("./customAttributePostModel");
const customAttributePutModel_1 = require("./customAttributePutModel");
const customAttributeSearchQueryModel_1 = require("./customAttributeSearchQueryModel");
const customAttributeTemplateModel_1 = require("./customAttributeTemplateModel");
const customAttributeTemplatePostModel_1 = require("./customAttributeTemplatePostModel");
const customAttributeTemplatePutModel_1 = require("./customAttributeTemplatePutModel");
const customAttributeTemplateSearchQueryModel_1 = require("./customAttributeTemplateSearchQueryModel");
const customAttributeTestPlanProjectRelationPutModel_1 = require("./customAttributeTestPlanProjectRelationPutModel");
const customAttributeTypesEnum_1 = require("./customAttributeTypesEnum");
const dateTimeRangeSelectorModel_1 = require("./dateTimeRangeSelectorModel");
const deletionState_1 = require("./deletionState");
const externalLinkModel_1 = require("./externalLinkModel");
const failureCategoryModel_1 = require("./failureCategoryModel");
const failureClassModel_1 = require("./failureClassModel");
const failureClassRegexModel_1 = require("./failureClassRegexModel");
const filterModel_1 = require("./filterModel");
const flakyBulkModel_1 = require("./flakyBulkModel");
const getXlsxTestPointsByTestPlanModel_1 = require("./getXlsxTestPointsByTestPlanModel");
const globalCustomAttributePostModel_1 = require("./globalCustomAttributePostModel");
const globalCustomAttributeUpdateModel_1 = require("./globalCustomAttributeUpdateModel");
const globalSearchItemResult_1 = require("./globalSearchItemResult");
const globalSearchRequest_1 = require("./globalSearchRequest");
const globalSearchResponse_1 = require("./globalSearchResponse");
const guidChangedFieldViewModel_1 = require("./guidChangedFieldViewModel");
const guidExtractionModel_1 = require("./guidExtractionModel");
const imageResizeType_1 = require("./imageResizeType");
const int32ChangedFieldViewModel_1 = require("./int32ChangedFieldViewModel");
const int32RangeSelectorModel_1 = require("./int32RangeSelectorModel");
const int64ChangedFieldViewModel_1 = require("./int64ChangedFieldViewModel");
const int64RangeSelectorModel_1 = require("./int64RangeSelectorModel");
const iterationModel_1 = require("./iterationModel");
const iterationPutModel_1 = require("./iterationPutModel");
const labelPostModel_1 = require("./labelPostModel");
const labelShortModel_1 = require("./labelShortModel");
const lastTestResultModel_1 = require("./lastTestResultModel");
const linkModel_1 = require("./linkModel");
const linkPostModel_1 = require("./linkPostModel");
const linkPutModel_1 = require("./linkPutModel");
const linkShortModel_1 = require("./linkShortModel");
const linkSubGetModel_1 = require("./linkSubGetModel");
const linkType_1 = require("./linkType");
const namedEntityModel_1 = require("./namedEntityModel");
const noContentResult_1 = require("./noContentResult");
const notificationModel_1 = require("./notificationModel");
const notificationQueryFilterModel_1 = require("./notificationQueryFilterModel");
const notificationTypeModel_1 = require("./notificationTypeModel");
const operation_1 = require("./operation");
const parameterFilterModel_1 = require("./parameterFilterModel");
const parameterGroupModel_1 = require("./parameterGroupModel");
const parameterIterationModel_1 = require("./parameterIterationModel");
const parameterModel_1 = require("./parameterModel");
const parameterPostModel_1 = require("./parameterPostModel");
const parameterPutModel_1 = require("./parameterPutModel");
const parameterShortModel_1 = require("./parameterShortModel");
const periodViewModel_1 = require("./periodViewModel");
const periodViewModelChangedFieldViewModel_1 = require("./periodViewModelChangedFieldViewModel");
const problemDetails_1 = require("./problemDetails");
const projectAttributesFilterModel_1 = require("./projectAttributesFilterModel");
const projectCustomAttributeTemplateGetModel_1 = require("./projectCustomAttributeTemplateGetModel");
const projectCustomAttributesTemplatesFilterModel_1 = require("./projectCustomAttributesTemplatesFilterModel");
const projectExportQueryModel_1 = require("./projectExportQueryModel");
const projectExportWithTestPlansPostModel_1 = require("./projectExportWithTestPlansPostModel");
const projectExtractionModel_1 = require("./projectExtractionModel");
const projectModel_1 = require("./projectModel");
const projectPostModel_1 = require("./projectPostModel");
const projectPutModel_1 = require("./projectPutModel");
const projectSelectModel_1 = require("./projectSelectModel");
const projectShortestModel_1 = require("./projectShortestModel");
const projectTestPlansFilterModel_1 = require("./projectTestPlansFilterModel");
const projectsFilterModel_1 = require("./projectsFilterModel");
const projectsFilterModelAutotestsCount_1 = require("./projectsFilterModelAutotestsCount");
const projectsFilterModelChecklistsCount_1 = require("./projectsFilterModelChecklistsCount");
const projectsFilterModelCreatedDate_1 = require("./projectsFilterModelCreatedDate");
const projectsFilterModelSharedStepsCount_1 = require("./projectsFilterModelSharedStepsCount");
const projectsFilterModelTestCasesCount_1 = require("./projectsFilterModelTestCasesCount");
const publicTestPointModel_1 = require("./publicTestPointModel");
const publicTestRunModel_1 = require("./publicTestRunModel");
const requestTypeModel_1 = require("./requestTypeModel");
const searchAutoTestsQueryIncludesModel_1 = require("./searchAutoTestsQueryIncludesModel");
const searchCustomAttributeTemplateGetModel_1 = require("./searchCustomAttributeTemplateGetModel");
const searchWebhooksQueryModel_1 = require("./searchWebhooksQueryModel");
const sectionModel_1 = require("./sectionModel");
const sectionMoveModel_1 = require("./sectionMoveModel");
const sectionPostModel_1 = require("./sectionPostModel");
const sectionPutModel_1 = require("./sectionPutModel");
const sectionRenameModel_1 = require("./sectionRenameModel");
const sectionSharedStep_1 = require("./sectionSharedStep");
const sectionWithStepsModel_1 = require("./sectionWithStepsModel");
const sharedStepChangeViewModel_1 = require("./sharedStepChangeViewModel");
const sharedStepModel_1 = require("./sharedStepModel");
const sharedStepReferenceModel_1 = require("./sharedStepReferenceModel");
const sharedStepReferenceSectionModel_1 = require("./sharedStepReferenceSectionModel");
const sharedStepReferenceSectionsQueryFilterModel_1 = require("./sharedStepReferenceSectionsQueryFilterModel");
const sharedStepReferenceSectionsQueryFilterModelCreatedDate_1 = require("./sharedStepReferenceSectionsQueryFilterModelCreatedDate");
const sharedStepReferenceSectionsQueryFilterModelModifiedDate_1 = require("./sharedStepReferenceSectionsQueryFilterModelModifiedDate");
const sharedStepReferencesQueryFilterModel_1 = require("./sharedStepReferencesQueryFilterModel");
const sharedStepResultModel_1 = require("./sharedStepResultModel");
const shortConfiguration_1 = require("./shortConfiguration");
const stepCommentModel_1 = require("./stepCommentModel");
const stepModel_1 = require("./stepModel");
const stepPostModel_1 = require("./stepPostModel");
const stepPutModel_1 = require("./stepPutModel");
const stepResultModel_1 = require("./stepResultModel");
const stringArrayChangedFieldViewModel_1 = require("./stringArrayChangedFieldViewModel");
const stringChangedFieldViewModel_1 = require("./stringChangedFieldViewModel");
const stringChangedFieldWithDiffsViewModel_1 = require("./stringChangedFieldWithDiffsViewModel");
const tagExtractionModel_1 = require("./tagExtractionModel");
const tagModel_1 = require("./tagModel");
const tagPostModel_1 = require("./tagPostModel");
const tagPutModel_1 = require("./tagPutModel");
const tagSelectModel_1 = require("./tagSelectModel");
const tagShortModel_1 = require("./tagShortModel");
const tagsFilterModel_1 = require("./tagsFilterModel");
const tagsFilterModelCreatedDate_1 = require("./tagsFilterModelCreatedDate");
const testPlanChangeModel_1 = require("./testPlanChangeModel");
const testPlanChangedFieldsViewModel_1 = require("./testPlanChangedFieldsViewModel");
const testPlanExtractionModel_1 = require("./testPlanExtractionModel");
const testPlanGroupByStatus_1 = require("./testPlanGroupByStatus");
const testPlanGroupByTestSuite_1 = require("./testPlanGroupByTestSuite");
const testPlanGroupByTester_1 = require("./testPlanGroupByTester");
const testPlanGroupByTesterAndStatus_1 = require("./testPlanGroupByTesterAndStatus");
const testPlanLink_1 = require("./testPlanLink");
const testPlanModel_1 = require("./testPlanModel");
const testPlanPostModel_1 = require("./testPlanPostModel");
const testPlanPutModel_1 = require("./testPlanPutModel");
const testPlanSelectModel_1 = require("./testPlanSelectModel");
const testPlanShortModel_1 = require("./testPlanShortModel");
const testPlanStatusModel_1 = require("./testPlanStatusModel");
const testPlanWithAnalyticModel_1 = require("./testPlanWithAnalyticModel");
const testPlanWithTestSuiteTreeModel_1 = require("./testPlanWithTestSuiteTreeModel");
const testPointAnalyticResult_1 = require("./testPointAnalyticResult");
const testPointByTestSuiteModel_1 = require("./testPointByTestSuiteModel");
const testPointChangeViewModel_1 = require("./testPointChangeViewModel");
const testPointChangeViewModelChangedFieldViewModel_1 = require("./testPointChangeViewModelChangedFieldViewModel");
const testPointFilterModel_1 = require("./testPointFilterModel");
const testPointFilterModelCreatedDate_1 = require("./testPointFilterModelCreatedDate");
const testPointFilterModelDuration_1 = require("./testPointFilterModelDuration");
const testPointFilterModelModifiedDate_1 = require("./testPointFilterModelModifiedDate");
const testPointFilterModelWorkItemCreatedDate_1 = require("./testPointFilterModelWorkItemCreatedDate");
const testPointFilterModelWorkItemMedianDuration_1 = require("./testPointFilterModelWorkItemMedianDuration");
const testPointFilterModelWorkItemModifiedDate_1 = require("./testPointFilterModelWorkItemModifiedDate");
const testPointPutModel_1 = require("./testPointPutModel");
const testPointRelatedToTestResult_1 = require("./testPointRelatedToTestResult");
const testPointResultModel_1 = require("./testPointResultModel");
const testPointSelectModel_1 = require("./testPointSelectModel");
const testPointSelector_1 = require("./testPointSelector");
const testPointShortGetModel_1 = require("./testPointShortGetModel");
const testPointShortModel_1 = require("./testPointShortModel");
const testPointStatus_1 = require("./testPointStatus");
const testPointWithLastResultModel_1 = require("./testPointWithLastResultModel");
const testPointsExtractionModel_1 = require("./testPointsExtractionModel");
const testPointsExtractionModelIds_1 = require("./testPointsExtractionModelIds");
const testResultChangeViewModel_1 = require("./testResultChangeViewModel");
const testResultChangeViewModelChangedFieldViewModel_1 = require("./testResultChangeViewModelChangedFieldViewModel");
const testResultChronologyModel_1 = require("./testResultChronologyModel");
const testResultConfiguration_1 = require("./testResultConfiguration");
const testResultHistoryReportModel_1 = require("./testResultHistoryReportModel");
const testResultModel_1 = require("./testResultModel");
const testResultOutcome_1 = require("./testResultOutcome");
const testResultShortGetModel_1 = require("./testResultShortGetModel");
const testResultShortModel_1 = require("./testResultShortModel");
const testResultStepCommentPutModel_1 = require("./testResultStepCommentPutModel");
const testResultUpdateModel_1 = require("./testResultUpdateModel");
const testResultV2GetModel_1 = require("./testResultV2GetModel");
const testResultV2ShortModel_1 = require("./testResultV2ShortModel");
const testResultsFilterModel_1 = require("./testResultsFilterModel");
const testResultsFilterModelCompletedOn_1 = require("./testResultsFilterModelCompletedOn");
const testResultsFilterModelCreatedDate_1 = require("./testResultsFilterModelCreatedDate");
const testResultsFilterModelDuration_1 = require("./testResultsFilterModelDuration");
const testResultsFilterModelModifiedDate_1 = require("./testResultsFilterModelModifiedDate");
const testResultsFilterModelStartedOn_1 = require("./testResultsFilterModelStartedOn");
const testResultsLocalFilterModel_1 = require("./testResultsLocalFilterModel");
const testResultsStatisticsGetModel_1 = require("./testResultsStatisticsGetModel");
const testResultsStatisticsGetModelFailureCategories_1 = require("./testResultsStatisticsGetModelFailureCategories");
const testResultsStatisticsGetModelStatuses_1 = require("./testResultsStatisticsGetModelStatuses");
const testRunAnalyticResultModel_1 = require("./testRunAnalyticResultModel");
const testRunExtractionModel_1 = require("./testRunExtractionModel");
const testRunExtractionModelIds_1 = require("./testRunExtractionModelIds");
const testRunFillByAutoTestsPostModel_1 = require("./testRunFillByAutoTestsPostModel");
const testRunFillByConfigurationsPostModel_1 = require("./testRunFillByConfigurationsPostModel");
const testRunFillByWorkItemsPostModel_1 = require("./testRunFillByWorkItemsPostModel");
const testRunFilterModel_1 = require("./testRunFilterModel");
const testRunFilterModelAutoTestsCount_1 = require("./testRunFilterModelAutoTestsCount");
const testRunFilterModelCompletedDate_1 = require("./testRunFilterModelCompletedDate");
const testRunFilterModelCreatedDate_1 = require("./testRunFilterModelCreatedDate");
const testRunFilterModelStartedDate_1 = require("./testRunFilterModelStartedDate");
const testRunGroupByFailureClassModel_1 = require("./testRunGroupByFailureClassModel");
const testRunGroupByStatusModel_1 = require("./testRunGroupByStatusModel");
const testRunModel_1 = require("./testRunModel");
const testRunSearchQueryModel_1 = require("./testRunSearchQueryModel");
const testRunSelectModel_1 = require("./testRunSelectModel");
const testRunShortGetModel_1 = require("./testRunShortGetModel");
const testRunShortModel_1 = require("./testRunShortModel");
const testRunState_1 = require("./testRunState");
const testRunStatisticsErrorCategoriesGetModel_1 = require("./testRunStatisticsErrorCategoriesGetModel");
const testRunStatisticsStatusesGetModel_1 = require("./testRunStatisticsStatusesGetModel");
const testRunTestResultsPartialBulkSetModel_1 = require("./testRunTestResultsPartialBulkSetModel");
const testRunTestResultsPartialBulkSetModelSelector_1 = require("./testRunTestResultsPartialBulkSetModelSelector");
const testRunTestResultsSelectModel_1 = require("./testRunTestResultsSelectModel");
const testRunTestResultsSelectModelFilter_1 = require("./testRunTestResultsSelectModelFilter");
const testRunTestResultsSelectModelTestResultIdsExtractionModel_1 = require("./testRunTestResultsSelectModelTestResultIdsExtractionModel");
const testRunUpdateMultipleModel_1 = require("./testRunUpdateMultipleModel");
const testRunV2GetModel_1 = require("./testRunV2GetModel");
const testRunV2PostShortModel_1 = require("./testRunV2PostShortModel");
const testRunV2PutModel_1 = require("./testRunV2PutModel");
const testSuiteChangeViewModel_1 = require("./testSuiteChangeViewModel");
const testSuiteChangeViewModelChangedFieldViewModel_1 = require("./testSuiteChangeViewModelChangedFieldViewModel");
const testSuiteType_1 = require("./testSuiteType");
const testSuiteV2GetModel_1 = require("./testSuiteV2GetModel");
const testSuiteV2PostModel_1 = require("./testSuiteV2PostModel");
const testSuiteV2PutModel_1 = require("./testSuiteV2PutModel");
const testSuiteV2TreeModel_1 = require("./testSuiteV2TreeModel");
const testSuiteWithChildrenModel_1 = require("./testSuiteWithChildrenModel");
const testSuiteWorkItemsSearchModel_1 = require("./testSuiteWorkItemsSearchModel");
const testSuiteWorkItemsSearchModelDuration_1 = require("./testSuiteWorkItemsSearchModelDuration");
const testSuiteWorkItemsSearchModelMedianDuration_1 = require("./testSuiteWorkItemsSearchModelMedianDuration");
const updateAttachmentShortModel_1 = require("./updateAttachmentShortModel");
const updateLinkShortModel_1 = require("./updateLinkShortModel");
const userRankModel_1 = require("./userRankModel");
const userWithRankModel_1 = require("./userWithRankModel");
const validateAntiForgeryTokenAttribute_1 = require("./validateAntiForgeryTokenAttribute");
const validationProblemDetails_1 = require("./validationProblemDetails");
const webHookEventType_1 = require("./webHookEventType");
const webHookEventTypeModel_1 = require("./webHookEventTypeModel");
const webHookLogModel_1 = require("./webHookLogModel");
const webHookModel_1 = require("./webHookModel");
const webHookPostModel_1 = require("./webHookPostModel");
const webHookTestModel_1 = require("./webHookTestModel");
const webhookResponse_1 = require("./webhookResponse");
const webhookVariablesType_1 = require("./webhookVariablesType");
const workItemChangeModel_1 = require("./workItemChangeModel");
const workItemChangedAttributeViewModel_1 = require("./workItemChangedAttributeViewModel");
const workItemChangedFieldsViewModel_1 = require("./workItemChangedFieldsViewModel");
const workItemCommentModel_1 = require("./workItemCommentModel");
const workItemCommentPostModel_1 = require("./workItemCommentPostModel");
const workItemCommentPutModel_1 = require("./workItemCommentPutModel");
const workItemEntityTypes_1 = require("./workItemEntityTypes");
const workItemExtractionModel_1 = require("./workItemExtractionModel");
const workItemExtractionModelIds_1 = require("./workItemExtractionModelIds");
const workItemExtractionModelSectionIds_1 = require("./workItemExtractionModelSectionIds");
const workItemFilterModel_1 = require("./workItemFilterModel");
const workItemGroupGetModel_1 = require("./workItemGroupGetModel");
const workItemGroupGetModelSelectModel_1 = require("./workItemGroupGetModelSelectModel");
const workItemGroupModel_1 = require("./workItemGroupModel");
const workItemGroupType_1 = require("./workItemGroupType");
const workItemIdModel_1 = require("./workItemIdModel");
const workItemIdentifierModel_1 = require("./workItemIdentifierModel");
const workItemLikeModel_1 = require("./workItemLikeModel");
const workItemLinkChangeViewModel_1 = require("./workItemLinkChangeViewModel");
const workItemLinkChangeViewModelArrayChangedFieldViewModel_1 = require("./workItemLinkChangeViewModelArrayChangedFieldViewModel");
const workItemLocalFilterModel_1 = require("./workItemLocalFilterModel");
const workItemLocalSelectModel_1 = require("./workItemLocalSelectModel");
const workItemLocalSelectModelExtractionModel_1 = require("./workItemLocalSelectModelExtractionModel");
const workItemLocalSelectModelFilter_1 = require("./workItemLocalSelectModelFilter");
const workItemModel_1 = require("./workItemModel");
const workItemMovePostModel_1 = require("./workItemMovePostModel");
const workItemPostModel_1 = require("./workItemPostModel");
const workItemPriorityModel_1 = require("./workItemPriorityModel");
const workItemPutModel_1 = require("./workItemPutModel");
const workItemSearchQueryModel_1 = require("./workItemSearchQueryModel");
const workItemSelectModel_1 = require("./workItemSelectModel");
const workItemShortModel_1 = require("./workItemShortModel");
const workItemState_1 = require("./workItemState");
const workItemStates_1 = require("./workItemStates");
const workItemStepChangeViewModel_1 = require("./workItemStepChangeViewModel");
const workItemStepChangeViewModelArrayChangedFieldWithDiffsViewModel_1 = require("./workItemStepChangeViewModelArrayChangedFieldWithDiffsViewModel");
const workItemVersionModel_1 = require("./workItemVersionModel");
let primitives = [
    "string",
    "boolean",
    "double",
    "integer",
    "long",
    "float",
    "number",
    "any"
];
let enumsMap = {
    "ActionUpdate": actionUpdate_1.ActionUpdate,
    "AutotestResultOutcome": autotestResultOutcome_1.AutotestResultOutcome,
    "AvailableTestResultOutcome": availableTestResultOutcome_1.AvailableTestResultOutcome,
    "BackgroundJobState": backgroundJobState_1.BackgroundJobState,
    "BackgroundJobType": backgroundJobType_1.BackgroundJobType,
    "CustomAttributeTypesEnum": customAttributeTypesEnum_1.CustomAttributeTypesEnum,
    "DeletionState": deletionState_1.DeletionState,
    "FailureCategoryModel": failureCategoryModel_1.FailureCategoryModel,
    "ImageResizeType": imageResizeType_1.ImageResizeType,
    "LinkType": linkType_1.LinkType,
    "NotificationTypeModel": notificationTypeModel_1.NotificationTypeModel,
    "RequestTypeModel": requestTypeModel_1.RequestTypeModel,
    "TestPlanStatusModel": testPlanStatusModel_1.TestPlanStatusModel,
    "TestPointStatus": testPointStatus_1.TestPointStatus,
    "TestResultOutcome": testResultOutcome_1.TestResultOutcome,
    "TestRunState": testRunState_1.TestRunState,
    "TestSuiteType": testSuiteType_1.TestSuiteType,
    "WebHookEventType": webHookEventType_1.WebHookEventType,
    "WebHookEventTypeModel": webHookEventTypeModel_1.WebHookEventTypeModel,
    "WebhookVariablesType": webhookVariablesType_1.WebhookVariablesType,
    "WorkItemEntityTypes": workItemEntityTypes_1.WorkItemEntityTypes,
    "WorkItemGroupType": workItemGroupType_1.WorkItemGroupType,
    "WorkItemPriorityModel": workItemPriorityModel_1.WorkItemPriorityModel,
    "WorkItemState": workItemState_1.WorkItemState,
    "WorkItemStates": workItemStates_1.WorkItemStates,
};
let typeMap = {
    "ApiV2AttachmentsPostRequest": apiV2AttachmentsPostRequest_1.ApiV2AttachmentsPostRequest,
    "AttachmentChangeViewModel": attachmentChangeViewModel_1.AttachmentChangeViewModel,
    "AttachmentChangeViewModelArrayChangedFieldViewModel": attachmentChangeViewModelArrayChangedFieldViewModel_1.AttachmentChangeViewModelArrayChangedFieldViewModel,
    "AttachmentModel": attachmentModel_1.AttachmentModel,
    "AttachmentModelAutoTestStepResultsModel": attachmentModelAutoTestStepResultsModel_1.AttachmentModelAutoTestStepResultsModel,
    "AttachmentPutModel": attachmentPutModel_1.AttachmentPutModel,
    "AttachmentPutModelAutoTestStepResultsModel": attachmentPutModelAutoTestStepResultsModel_1.AttachmentPutModelAutoTestStepResultsModel,
    "AutoTestAverageDurationModel": autoTestAverageDurationModel_1.AutoTestAverageDurationModel,
    "AutoTestChangeViewModel": autoTestChangeViewModel_1.AutoTestChangeViewModel,
    "AutoTestChangeViewModelArrayChangedFieldViewModel": autoTestChangeViewModelArrayChangedFieldViewModel_1.AutoTestChangeViewModelArrayChangedFieldViewModel,
    "AutoTestIdModel": autoTestIdModel_1.AutoTestIdModel,
    "AutoTestModel": autoTestModel_1.AutoTestModel,
    "AutoTestModelV2GetModel": autoTestModelV2GetModel_1.AutoTestModelV2GetModel,
    "AutoTestNamespaceModel": autoTestNamespaceModel_1.AutoTestNamespaceModel,
    "AutoTestPostModel": autoTestPostModel_1.AutoTestPostModel,
    "AutoTestPutModel": autoTestPutModel_1.AutoTestPutModel,
    "AutoTestRelatedToTestResult": autoTestRelatedToTestResult_1.AutoTestRelatedToTestResult,
    "AutoTestResultsForTestRunModel": autoTestResultsForTestRunModel_1.AutoTestResultsForTestRunModel,
    "AutoTestShortModel": autoTestShortModel_1.AutoTestShortModel,
    "AutoTestStepModel": autoTestStepModel_1.AutoTestStepModel,
    "AutotestFilterModel": autotestFilterModel_1.AutotestFilterModel,
    "AutotestFilterModelCreatedDate": autotestFilterModelCreatedDate_1.AutotestFilterModelCreatedDate,
    "AutotestFilterModelModifiedDate": autotestFilterModelModifiedDate_1.AutotestFilterModelModifiedDate,
    "AutotestFilterModelStabilityPercentage": autotestFilterModelStabilityPercentage_1.AutotestFilterModelStabilityPercentage,
    "AutotestHistoricalResultSelectModel": autotestHistoricalResultSelectModel_1.AutotestHistoricalResultSelectModel,
    "AutotestResultHistoricalGetModel": autotestResultHistoricalGetModel_1.AutotestResultHistoricalGetModel,
    "AutotestResultReasonSubGetModel": autotestResultReasonSubGetModel_1.AutotestResultReasonSubGetModel,
    "AutotestSelectModel": autotestSelectModel_1.AutotestSelectModel,
    "AutotestsExtractionModel": autotestsExtractionModel_1.AutotestsExtractionModel,
    "AutotestsExtractionModelIds": autotestsExtractionModelIds_1.AutotestsExtractionModelIds,
    "AutotestsSelectModel": autotestsSelectModel_1.AutotestsSelectModel,
    "BackgroundJobAttachmentModel": backgroundJobAttachmentModel_1.BackgroundJobAttachmentModel,
    "BackgroundJobFilterModel": backgroundJobFilterModel_1.BackgroundJobFilterModel,
    "BackgroundJobGetModel": backgroundJobGetModel_1.BackgroundJobGetModel,
    "BooleanChangedFieldViewModel": booleanChangedFieldViewModel_1.BooleanChangedFieldViewModel,
    "BooleanNullableChangedFieldViewModel": booleanNullableChangedFieldViewModel_1.BooleanNullableChangedFieldViewModel,
    "ConfigurationByParametersModel": configurationByParametersModel_1.ConfigurationByParametersModel,
    "ConfigurationExtractionModel": configurationExtractionModel_1.ConfigurationExtractionModel,
    "ConfigurationExtractionModelIds": configurationExtractionModelIds_1.ConfigurationExtractionModelIds,
    "ConfigurationExtractionModelProjectIds": configurationExtractionModelProjectIds_1.ConfigurationExtractionModelProjectIds,
    "ConfigurationFilterModel": configurationFilterModel_1.ConfigurationFilterModel,
    "ConfigurationModel": configurationModel_1.ConfigurationModel,
    "ConfigurationPostModel": configurationPostModel_1.ConfigurationPostModel,
    "ConfigurationPutModel": configurationPutModel_1.ConfigurationPutModel,
    "ConfigurationSelectModel": configurationSelectModel_1.ConfigurationSelectModel,
    "ConfigurationSelectModelExtractionModel": configurationSelectModelExtractionModel_1.ConfigurationSelectModelExtractionModel,
    "ConfigurationSelectModelFilter": configurationSelectModelFilter_1.ConfigurationSelectModelFilter,
    "CustomAttributeChangeModel": customAttributeChangeModel_1.CustomAttributeChangeModel,
    "CustomAttributeGetModel": customAttributeGetModel_1.CustomAttributeGetModel,
    "CustomAttributeModel": customAttributeModel_1.CustomAttributeModel,
    "CustomAttributeOptionModel": customAttributeOptionModel_1.CustomAttributeOptionModel,
    "CustomAttributeOptionPostModel": customAttributeOptionPostModel_1.CustomAttributeOptionPostModel,
    "CustomAttributePostModel": customAttributePostModel_1.CustomAttributePostModel,
    "CustomAttributePutModel": customAttributePutModel_1.CustomAttributePutModel,
    "CustomAttributeSearchQueryModel": customAttributeSearchQueryModel_1.CustomAttributeSearchQueryModel,
    "CustomAttributeTemplateModel": customAttributeTemplateModel_1.CustomAttributeTemplateModel,
    "CustomAttributeTemplatePostModel": customAttributeTemplatePostModel_1.CustomAttributeTemplatePostModel,
    "CustomAttributeTemplatePutModel": customAttributeTemplatePutModel_1.CustomAttributeTemplatePutModel,
    "CustomAttributeTemplateSearchQueryModel": customAttributeTemplateSearchQueryModel_1.CustomAttributeTemplateSearchQueryModel,
    "CustomAttributeTestPlanProjectRelationPutModel": customAttributeTestPlanProjectRelationPutModel_1.CustomAttributeTestPlanProjectRelationPutModel,
    "DateTimeRangeSelectorModel": dateTimeRangeSelectorModel_1.DateTimeRangeSelectorModel,
    "ExternalLinkModel": externalLinkModel_1.ExternalLinkModel,
    "FailureClassModel": failureClassModel_1.FailureClassModel,
    "FailureClassRegexModel": failureClassRegexModel_1.FailureClassRegexModel,
    "FilterModel": filterModel_1.FilterModel,
    "FlakyBulkModel": flakyBulkModel_1.FlakyBulkModel,
    "GetXlsxTestPointsByTestPlanModel": getXlsxTestPointsByTestPlanModel_1.GetXlsxTestPointsByTestPlanModel,
    "GlobalCustomAttributePostModel": globalCustomAttributePostModel_1.GlobalCustomAttributePostModel,
    "GlobalCustomAttributeUpdateModel": globalCustomAttributeUpdateModel_1.GlobalCustomAttributeUpdateModel,
    "GlobalSearchItemResult": globalSearchItemResult_1.GlobalSearchItemResult,
    "GlobalSearchRequest": globalSearchRequest_1.GlobalSearchRequest,
    "GlobalSearchResponse": globalSearchResponse_1.GlobalSearchResponse,
    "GuidChangedFieldViewModel": guidChangedFieldViewModel_1.GuidChangedFieldViewModel,
    "GuidExtractionModel": guidExtractionModel_1.GuidExtractionModel,
    "Int32ChangedFieldViewModel": int32ChangedFieldViewModel_1.Int32ChangedFieldViewModel,
    "Int32RangeSelectorModel": int32RangeSelectorModel_1.Int32RangeSelectorModel,
    "Int64ChangedFieldViewModel": int64ChangedFieldViewModel_1.Int64ChangedFieldViewModel,
    "Int64RangeSelectorModel": int64RangeSelectorModel_1.Int64RangeSelectorModel,
    "IterationModel": iterationModel_1.IterationModel,
    "IterationPutModel": iterationPutModel_1.IterationPutModel,
    "LabelPostModel": labelPostModel_1.LabelPostModel,
    "LabelShortModel": labelShortModel_1.LabelShortModel,
    "LastTestResultModel": lastTestResultModel_1.LastTestResultModel,
    "LinkModel": linkModel_1.LinkModel,
    "LinkPostModel": linkPostModel_1.LinkPostModel,
    "LinkPutModel": linkPutModel_1.LinkPutModel,
    "LinkShortModel": linkShortModel_1.LinkShortModel,
    "LinkSubGetModel": linkSubGetModel_1.LinkSubGetModel,
    "NamedEntityModel": namedEntityModel_1.NamedEntityModel,
    "NoContentResult": noContentResult_1.NoContentResult,
    "NotificationModel": notificationModel_1.NotificationModel,
    "NotificationQueryFilterModel": notificationQueryFilterModel_1.NotificationQueryFilterModel,
    "Operation": operation_1.Operation,
    "ParameterFilterModel": parameterFilterModel_1.ParameterFilterModel,
    "ParameterGroupModel": parameterGroupModel_1.ParameterGroupModel,
    "ParameterIterationModel": parameterIterationModel_1.ParameterIterationModel,
    "ParameterModel": parameterModel_1.ParameterModel,
    "ParameterPostModel": parameterPostModel_1.ParameterPostModel,
    "ParameterPutModel": parameterPutModel_1.ParameterPutModel,
    "ParameterShortModel": parameterShortModel_1.ParameterShortModel,
    "PeriodViewModel": periodViewModel_1.PeriodViewModel,
    "PeriodViewModelChangedFieldViewModel": periodViewModelChangedFieldViewModel_1.PeriodViewModelChangedFieldViewModel,
    "ProblemDetails": problemDetails_1.ProblemDetails,
    "ProjectAttributesFilterModel": projectAttributesFilterModel_1.ProjectAttributesFilterModel,
    "ProjectCustomAttributeTemplateGetModel": projectCustomAttributeTemplateGetModel_1.ProjectCustomAttributeTemplateGetModel,
    "ProjectCustomAttributesTemplatesFilterModel": projectCustomAttributesTemplatesFilterModel_1.ProjectCustomAttributesTemplatesFilterModel,
    "ProjectExportQueryModel": projectExportQueryModel_1.ProjectExportQueryModel,
    "ProjectExportWithTestPlansPostModel": projectExportWithTestPlansPostModel_1.ProjectExportWithTestPlansPostModel,
    "ProjectExtractionModel": projectExtractionModel_1.ProjectExtractionModel,
    "ProjectModel": projectModel_1.ProjectModel,
    "ProjectPostModel": projectPostModel_1.ProjectPostModel,
    "ProjectPutModel": projectPutModel_1.ProjectPutModel,
    "ProjectSelectModel": projectSelectModel_1.ProjectSelectModel,
    "ProjectShortestModel": projectShortestModel_1.ProjectShortestModel,
    "ProjectTestPlansFilterModel": projectTestPlansFilterModel_1.ProjectTestPlansFilterModel,
    "ProjectsFilterModel": projectsFilterModel_1.ProjectsFilterModel,
    "ProjectsFilterModelAutotestsCount": projectsFilterModelAutotestsCount_1.ProjectsFilterModelAutotestsCount,
    "ProjectsFilterModelChecklistsCount": projectsFilterModelChecklistsCount_1.ProjectsFilterModelChecklistsCount,
    "ProjectsFilterModelCreatedDate": projectsFilterModelCreatedDate_1.ProjectsFilterModelCreatedDate,
    "ProjectsFilterModelSharedStepsCount": projectsFilterModelSharedStepsCount_1.ProjectsFilterModelSharedStepsCount,
    "ProjectsFilterModelTestCasesCount": projectsFilterModelTestCasesCount_1.ProjectsFilterModelTestCasesCount,
    "PublicTestPointModel": publicTestPointModel_1.PublicTestPointModel,
    "PublicTestRunModel": publicTestRunModel_1.PublicTestRunModel,
    "SearchAutoTestsQueryIncludesModel": searchAutoTestsQueryIncludesModel_1.SearchAutoTestsQueryIncludesModel,
    "SearchCustomAttributeTemplateGetModel": searchCustomAttributeTemplateGetModel_1.SearchCustomAttributeTemplateGetModel,
    "SearchWebhooksQueryModel": searchWebhooksQueryModel_1.SearchWebhooksQueryModel,
    "SectionModel": sectionModel_1.SectionModel,
    "SectionMoveModel": sectionMoveModel_1.SectionMoveModel,
    "SectionPostModel": sectionPostModel_1.SectionPostModel,
    "SectionPutModel": sectionPutModel_1.SectionPutModel,
    "SectionRenameModel": sectionRenameModel_1.SectionRenameModel,
    "SectionSharedStep": sectionSharedStep_1.SectionSharedStep,
    "SectionWithStepsModel": sectionWithStepsModel_1.SectionWithStepsModel,
    "SharedStepChangeViewModel": sharedStepChangeViewModel_1.SharedStepChangeViewModel,
    "SharedStepModel": sharedStepModel_1.SharedStepModel,
    "SharedStepReferenceModel": sharedStepReferenceModel_1.SharedStepReferenceModel,
    "SharedStepReferenceSectionModel": sharedStepReferenceSectionModel_1.SharedStepReferenceSectionModel,
    "SharedStepReferenceSectionsQueryFilterModel": sharedStepReferenceSectionsQueryFilterModel_1.SharedStepReferenceSectionsQueryFilterModel,
    "SharedStepReferenceSectionsQueryFilterModelCreatedDate": sharedStepReferenceSectionsQueryFilterModelCreatedDate_1.SharedStepReferenceSectionsQueryFilterModelCreatedDate,
    "SharedStepReferenceSectionsQueryFilterModelModifiedDate": sharedStepReferenceSectionsQueryFilterModelModifiedDate_1.SharedStepReferenceSectionsQueryFilterModelModifiedDate,
    "SharedStepReferencesQueryFilterModel": sharedStepReferencesQueryFilterModel_1.SharedStepReferencesQueryFilterModel,
    "SharedStepResultModel": sharedStepResultModel_1.SharedStepResultModel,
    "ShortConfiguration": shortConfiguration_1.ShortConfiguration,
    "StepCommentModel": stepCommentModel_1.StepCommentModel,
    "StepModel": stepModel_1.StepModel,
    "StepPostModel": stepPostModel_1.StepPostModel,
    "StepPutModel": stepPutModel_1.StepPutModel,
    "StepResultModel": stepResultModel_1.StepResultModel,
    "StringArrayChangedFieldViewModel": stringArrayChangedFieldViewModel_1.StringArrayChangedFieldViewModel,
    "StringChangedFieldViewModel": stringChangedFieldViewModel_1.StringChangedFieldViewModel,
    "StringChangedFieldWithDiffsViewModel": stringChangedFieldWithDiffsViewModel_1.StringChangedFieldWithDiffsViewModel,
    "TagExtractionModel": tagExtractionModel_1.TagExtractionModel,
    "TagModel": tagModel_1.TagModel,
    "TagPostModel": tagPostModel_1.TagPostModel,
    "TagPutModel": tagPutModel_1.TagPutModel,
    "TagSelectModel": tagSelectModel_1.TagSelectModel,
    "TagShortModel": tagShortModel_1.TagShortModel,
    "TagsFilterModel": tagsFilterModel_1.TagsFilterModel,
    "TagsFilterModelCreatedDate": tagsFilterModelCreatedDate_1.TagsFilterModelCreatedDate,
    "TestPlanChangeModel": testPlanChangeModel_1.TestPlanChangeModel,
    "TestPlanChangedFieldsViewModel": testPlanChangedFieldsViewModel_1.TestPlanChangedFieldsViewModel,
    "TestPlanExtractionModel": testPlanExtractionModel_1.TestPlanExtractionModel,
    "TestPlanGroupByStatus": testPlanGroupByStatus_1.TestPlanGroupByStatus,
    "TestPlanGroupByTestSuite": testPlanGroupByTestSuite_1.TestPlanGroupByTestSuite,
    "TestPlanGroupByTester": testPlanGroupByTester_1.TestPlanGroupByTester,
    "TestPlanGroupByTesterAndStatus": testPlanGroupByTesterAndStatus_1.TestPlanGroupByTesterAndStatus,
    "TestPlanLink": testPlanLink_1.TestPlanLink,
    "TestPlanModel": testPlanModel_1.TestPlanModel,
    "TestPlanPostModel": testPlanPostModel_1.TestPlanPostModel,
    "TestPlanPutModel": testPlanPutModel_1.TestPlanPutModel,
    "TestPlanSelectModel": testPlanSelectModel_1.TestPlanSelectModel,
    "TestPlanShortModel": testPlanShortModel_1.TestPlanShortModel,
    "TestPlanWithAnalyticModel": testPlanWithAnalyticModel_1.TestPlanWithAnalyticModel,
    "TestPlanWithTestSuiteTreeModel": testPlanWithTestSuiteTreeModel_1.TestPlanWithTestSuiteTreeModel,
    "TestPointAnalyticResult": testPointAnalyticResult_1.TestPointAnalyticResult,
    "TestPointByTestSuiteModel": testPointByTestSuiteModel_1.TestPointByTestSuiteModel,
    "TestPointChangeViewModel": testPointChangeViewModel_1.TestPointChangeViewModel,
    "TestPointChangeViewModelChangedFieldViewModel": testPointChangeViewModelChangedFieldViewModel_1.TestPointChangeViewModelChangedFieldViewModel,
    "TestPointFilterModel": testPointFilterModel_1.TestPointFilterModel,
    "TestPointFilterModelCreatedDate": testPointFilterModelCreatedDate_1.TestPointFilterModelCreatedDate,
    "TestPointFilterModelDuration": testPointFilterModelDuration_1.TestPointFilterModelDuration,
    "TestPointFilterModelModifiedDate": testPointFilterModelModifiedDate_1.TestPointFilterModelModifiedDate,
    "TestPointFilterModelWorkItemCreatedDate": testPointFilterModelWorkItemCreatedDate_1.TestPointFilterModelWorkItemCreatedDate,
    "TestPointFilterModelWorkItemMedianDuration": testPointFilterModelWorkItemMedianDuration_1.TestPointFilterModelWorkItemMedianDuration,
    "TestPointFilterModelWorkItemModifiedDate": testPointFilterModelWorkItemModifiedDate_1.TestPointFilterModelWorkItemModifiedDate,
    "TestPointPutModel": testPointPutModel_1.TestPointPutModel,
    "TestPointRelatedToTestResult": testPointRelatedToTestResult_1.TestPointRelatedToTestResult,
    "TestPointResultModel": testPointResultModel_1.TestPointResultModel,
    "TestPointSelectModel": testPointSelectModel_1.TestPointSelectModel,
    "TestPointSelector": testPointSelector_1.TestPointSelector,
    "TestPointShortGetModel": testPointShortGetModel_1.TestPointShortGetModel,
    "TestPointShortModel": testPointShortModel_1.TestPointShortModel,
    "TestPointWithLastResultModel": testPointWithLastResultModel_1.TestPointWithLastResultModel,
    "TestPointsExtractionModel": testPointsExtractionModel_1.TestPointsExtractionModel,
    "TestPointsExtractionModelIds": testPointsExtractionModelIds_1.TestPointsExtractionModelIds,
    "TestResultChangeViewModel": testResultChangeViewModel_1.TestResultChangeViewModel,
    "TestResultChangeViewModelChangedFieldViewModel": testResultChangeViewModelChangedFieldViewModel_1.TestResultChangeViewModelChangedFieldViewModel,
    "TestResultChronologyModel": testResultChronologyModel_1.TestResultChronologyModel,
    "TestResultConfiguration": testResultConfiguration_1.TestResultConfiguration,
    "TestResultHistoryReportModel": testResultHistoryReportModel_1.TestResultHistoryReportModel,
    "TestResultModel": testResultModel_1.TestResultModel,
    "TestResultShortGetModel": testResultShortGetModel_1.TestResultShortGetModel,
    "TestResultShortModel": testResultShortModel_1.TestResultShortModel,
    "TestResultStepCommentPutModel": testResultStepCommentPutModel_1.TestResultStepCommentPutModel,
    "TestResultUpdateModel": testResultUpdateModel_1.TestResultUpdateModel,
    "TestResultV2GetModel": testResultV2GetModel_1.TestResultV2GetModel,
    "TestResultV2ShortModel": testResultV2ShortModel_1.TestResultV2ShortModel,
    "TestResultsFilterModel": testResultsFilterModel_1.TestResultsFilterModel,
    "TestResultsFilterModelCompletedOn": testResultsFilterModelCompletedOn_1.TestResultsFilterModelCompletedOn,
    "TestResultsFilterModelCreatedDate": testResultsFilterModelCreatedDate_1.TestResultsFilterModelCreatedDate,
    "TestResultsFilterModelDuration": testResultsFilterModelDuration_1.TestResultsFilterModelDuration,
    "TestResultsFilterModelModifiedDate": testResultsFilterModelModifiedDate_1.TestResultsFilterModelModifiedDate,
    "TestResultsFilterModelStartedOn": testResultsFilterModelStartedOn_1.TestResultsFilterModelStartedOn,
    "TestResultsLocalFilterModel": testResultsLocalFilterModel_1.TestResultsLocalFilterModel,
    "TestResultsStatisticsGetModel": testResultsStatisticsGetModel_1.TestResultsStatisticsGetModel,
    "TestResultsStatisticsGetModelFailureCategories": testResultsStatisticsGetModelFailureCategories_1.TestResultsStatisticsGetModelFailureCategories,
    "TestResultsStatisticsGetModelStatuses": testResultsStatisticsGetModelStatuses_1.TestResultsStatisticsGetModelStatuses,
    "TestRunAnalyticResultModel": testRunAnalyticResultModel_1.TestRunAnalyticResultModel,
    "TestRunExtractionModel": testRunExtractionModel_1.TestRunExtractionModel,
    "TestRunExtractionModelIds": testRunExtractionModelIds_1.TestRunExtractionModelIds,
    "TestRunFillByAutoTestsPostModel": testRunFillByAutoTestsPostModel_1.TestRunFillByAutoTestsPostModel,
    "TestRunFillByConfigurationsPostModel": testRunFillByConfigurationsPostModel_1.TestRunFillByConfigurationsPostModel,
    "TestRunFillByWorkItemsPostModel": testRunFillByWorkItemsPostModel_1.TestRunFillByWorkItemsPostModel,
    "TestRunFilterModel": testRunFilterModel_1.TestRunFilterModel,
    "TestRunFilterModelAutoTestsCount": testRunFilterModelAutoTestsCount_1.TestRunFilterModelAutoTestsCount,
    "TestRunFilterModelCompletedDate": testRunFilterModelCompletedDate_1.TestRunFilterModelCompletedDate,
    "TestRunFilterModelCreatedDate": testRunFilterModelCreatedDate_1.TestRunFilterModelCreatedDate,
    "TestRunFilterModelStartedDate": testRunFilterModelStartedDate_1.TestRunFilterModelStartedDate,
    "TestRunGroupByFailureClassModel": testRunGroupByFailureClassModel_1.TestRunGroupByFailureClassModel,
    "TestRunGroupByStatusModel": testRunGroupByStatusModel_1.TestRunGroupByStatusModel,
    "TestRunModel": testRunModel_1.TestRunModel,
    "TestRunSearchQueryModel": testRunSearchQueryModel_1.TestRunSearchQueryModel,
    "TestRunSelectModel": testRunSelectModel_1.TestRunSelectModel,
    "TestRunShortGetModel": testRunShortGetModel_1.TestRunShortGetModel,
    "TestRunShortModel": testRunShortModel_1.TestRunShortModel,
    "TestRunStatisticsErrorCategoriesGetModel": testRunStatisticsErrorCategoriesGetModel_1.TestRunStatisticsErrorCategoriesGetModel,
    "TestRunStatisticsStatusesGetModel": testRunStatisticsStatusesGetModel_1.TestRunStatisticsStatusesGetModel,
    "TestRunTestResultsPartialBulkSetModel": testRunTestResultsPartialBulkSetModel_1.TestRunTestResultsPartialBulkSetModel,
    "TestRunTestResultsPartialBulkSetModelSelector": testRunTestResultsPartialBulkSetModelSelector_1.TestRunTestResultsPartialBulkSetModelSelector,
    "TestRunTestResultsSelectModel": testRunTestResultsSelectModel_1.TestRunTestResultsSelectModel,
    "TestRunTestResultsSelectModelFilter": testRunTestResultsSelectModelFilter_1.TestRunTestResultsSelectModelFilter,
    "TestRunTestResultsSelectModelTestResultIdsExtractionModel": testRunTestResultsSelectModelTestResultIdsExtractionModel_1.TestRunTestResultsSelectModelTestResultIdsExtractionModel,
    "TestRunUpdateMultipleModel": testRunUpdateMultipleModel_1.TestRunUpdateMultipleModel,
    "TestRunV2GetModel": testRunV2GetModel_1.TestRunV2GetModel,
    "TestRunV2PostShortModel": testRunV2PostShortModel_1.TestRunV2PostShortModel,
    "TestRunV2PutModel": testRunV2PutModel_1.TestRunV2PutModel,
    "TestSuiteChangeViewModel": testSuiteChangeViewModel_1.TestSuiteChangeViewModel,
    "TestSuiteChangeViewModelChangedFieldViewModel": testSuiteChangeViewModelChangedFieldViewModel_1.TestSuiteChangeViewModelChangedFieldViewModel,
    "TestSuiteV2GetModel": testSuiteV2GetModel_1.TestSuiteV2GetModel,
    "TestSuiteV2PostModel": testSuiteV2PostModel_1.TestSuiteV2PostModel,
    "TestSuiteV2PutModel": testSuiteV2PutModel_1.TestSuiteV2PutModel,
    "TestSuiteV2TreeModel": testSuiteV2TreeModel_1.TestSuiteV2TreeModel,
    "TestSuiteWithChildrenModel": testSuiteWithChildrenModel_1.TestSuiteWithChildrenModel,
    "TestSuiteWorkItemsSearchModel": testSuiteWorkItemsSearchModel_1.TestSuiteWorkItemsSearchModel,
    "TestSuiteWorkItemsSearchModelDuration": testSuiteWorkItemsSearchModelDuration_1.TestSuiteWorkItemsSearchModelDuration,
    "TestSuiteWorkItemsSearchModelMedianDuration": testSuiteWorkItemsSearchModelMedianDuration_1.TestSuiteWorkItemsSearchModelMedianDuration,
    "UpdateAttachmentShortModel": updateAttachmentShortModel_1.UpdateAttachmentShortModel,
    "UpdateLinkShortModel": updateLinkShortModel_1.UpdateLinkShortModel,
    "UserRankModel": userRankModel_1.UserRankModel,
    "UserWithRankModel": userWithRankModel_1.UserWithRankModel,
    "ValidateAntiForgeryTokenAttribute": validateAntiForgeryTokenAttribute_1.ValidateAntiForgeryTokenAttribute,
    "ValidationProblemDetails": validationProblemDetails_1.ValidationProblemDetails,
    "WebHookLogModel": webHookLogModel_1.WebHookLogModel,
    "WebHookModel": webHookModel_1.WebHookModel,
    "WebHookPostModel": webHookPostModel_1.WebHookPostModel,
    "WebHookTestModel": webHookTestModel_1.WebHookTestModel,
    "WebhookResponse": webhookResponse_1.WebhookResponse,
    "WorkItemChangeModel": workItemChangeModel_1.WorkItemChangeModel,
    "WorkItemChangedAttributeViewModel": workItemChangedAttributeViewModel_1.WorkItemChangedAttributeViewModel,
    "WorkItemChangedFieldsViewModel": workItemChangedFieldsViewModel_1.WorkItemChangedFieldsViewModel,
    "WorkItemCommentModel": workItemCommentModel_1.WorkItemCommentModel,
    "WorkItemCommentPostModel": workItemCommentPostModel_1.WorkItemCommentPostModel,
    "WorkItemCommentPutModel": workItemCommentPutModel_1.WorkItemCommentPutModel,
    "WorkItemExtractionModel": workItemExtractionModel_1.WorkItemExtractionModel,
    "WorkItemExtractionModelIds": workItemExtractionModelIds_1.WorkItemExtractionModelIds,
    "WorkItemExtractionModelSectionIds": workItemExtractionModelSectionIds_1.WorkItemExtractionModelSectionIds,
    "WorkItemFilterModel": workItemFilterModel_1.WorkItemFilterModel,
    "WorkItemGroupGetModel": workItemGroupGetModel_1.WorkItemGroupGetModel,
    "WorkItemGroupGetModelSelectModel": workItemGroupGetModelSelectModel_1.WorkItemGroupGetModelSelectModel,
    "WorkItemGroupModel": workItemGroupModel_1.WorkItemGroupModel,
    "WorkItemIdModel": workItemIdModel_1.WorkItemIdModel,
    "WorkItemIdentifierModel": workItemIdentifierModel_1.WorkItemIdentifierModel,
    "WorkItemLikeModel": workItemLikeModel_1.WorkItemLikeModel,
    "WorkItemLinkChangeViewModel": workItemLinkChangeViewModel_1.WorkItemLinkChangeViewModel,
    "WorkItemLinkChangeViewModelArrayChangedFieldViewModel": workItemLinkChangeViewModelArrayChangedFieldViewModel_1.WorkItemLinkChangeViewModelArrayChangedFieldViewModel,
    "WorkItemLocalFilterModel": workItemLocalFilterModel_1.WorkItemLocalFilterModel,
    "WorkItemLocalSelectModel": workItemLocalSelectModel_1.WorkItemLocalSelectModel,
    "WorkItemLocalSelectModelExtractionModel": workItemLocalSelectModelExtractionModel_1.WorkItemLocalSelectModelExtractionModel,
    "WorkItemLocalSelectModelFilter": workItemLocalSelectModelFilter_1.WorkItemLocalSelectModelFilter,
    "WorkItemModel": workItemModel_1.WorkItemModel,
    "WorkItemMovePostModel": workItemMovePostModel_1.WorkItemMovePostModel,
    "WorkItemPostModel": workItemPostModel_1.WorkItemPostModel,
    "WorkItemPutModel": workItemPutModel_1.WorkItemPutModel,
    "WorkItemSearchQueryModel": workItemSearchQueryModel_1.WorkItemSearchQueryModel,
    "WorkItemSelectModel": workItemSelectModel_1.WorkItemSelectModel,
    "WorkItemShortModel": workItemShortModel_1.WorkItemShortModel,
    "WorkItemStepChangeViewModel": workItemStepChangeViewModel_1.WorkItemStepChangeViewModel,
    "WorkItemStepChangeViewModelArrayChangedFieldWithDiffsViewModel": workItemStepChangeViewModelArrayChangedFieldWithDiffsViewModel_1.WorkItemStepChangeViewModelArrayChangedFieldWithDiffsViewModel,
    "WorkItemVersionModel": workItemVersionModel_1.WorkItemVersionModel,
};
class ObjectSerializer {
    static findCorrectType(data, expectedType) {
        if (data == undefined) {
            return expectedType;
        }
        else if (primitives.indexOf(expectedType.toLowerCase()) !== -1) {
            return expectedType;
        }
        else if (expectedType === "Date") {
            return expectedType;
        }
        else {
            if (enumsMap[expectedType]) {
                return expectedType;
            }
            if (!typeMap[expectedType]) {
                return expectedType;
            }
            let discriminatorProperty = typeMap[expectedType].discriminator;
            if (discriminatorProperty == null) {
                return expectedType;
            }
            else {
                if (data[discriminatorProperty]) {
                    var discriminatorType = data[discriminatorProperty];
                    if (typeMap[discriminatorType]) {
                        return discriminatorType;
                    }
                    else {
                        return expectedType;
                    }
                }
                else {
                    return expectedType;
                }
            }
        }
    }
    static serialize(data, type) {
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) {
            let subType = type.replace("Array<", "");
            subType = subType.substring(0, subType.length - 1);
            let transformedData = [];
            for (let index = 0; index < data.length; index++) {
                let datum = data[index];
                transformedData.push(ObjectSerializer.serialize(datum, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return data.toISOString();
        }
        else {
            if (enumsMap[type]) {
                return data;
            }
            if (!typeMap[type]) {
                return data;
            }
            type = this.findCorrectType(data, type);
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            let instance = {};
            for (let index = 0; index < attributeTypes.length; index++) {
                let attributeType = attributeTypes[index];
                instance[attributeType.baseName] = ObjectSerializer.serialize(data[attributeType.name], attributeType.type);
            }
            return instance;
        }
    }
    static deserialize(data, type) {
        type = ObjectSerializer.findCorrectType(data, type);
        if (data == undefined) {
            return data;
        }
        else if (primitives.indexOf(type.toLowerCase()) !== -1) {
            return data;
        }
        else if (type.lastIndexOf("Array<", 0) === 0) {
            let subType = type.replace("Array<", "");
            subType = subType.substring(0, subType.length - 1);
            let transformedData = [];
            for (let index = 0; index < data.length; index++) {
                let datum = data[index];
                transformedData.push(ObjectSerializer.deserialize(datum, subType));
            }
            return transformedData;
        }
        else if (type === "Date") {
            return new Date(data);
        }
        else {
            if (enumsMap[type]) {
                return data;
            }
            if (!typeMap[type]) {
                return data;
            }
            let instance = new typeMap[type]();
            let attributeTypes = typeMap[type].getAttributeTypeMap();
            for (let index = 0; index < attributeTypes.length; index++) {
                let attributeType = attributeTypes[index];
                instance[attributeType.name] = ObjectSerializer.deserialize(data[attributeType.baseName], attributeType.type);
            }
            return instance;
        }
    }
}
exports.ObjectSerializer = ObjectSerializer;
class HttpBasicAuth {
    constructor() {
        this.username = '';
        this.password = '';
    }
    applyToRequest(requestOptions) {
        requestOptions.auth = {
            username: this.username, password: this.password
        };
    }
}
exports.HttpBasicAuth = HttpBasicAuth;
class HttpBearerAuth {
    constructor() {
        this.accessToken = '';
    }
    applyToRequest(requestOptions) {
        if (requestOptions && requestOptions.headers) {
            const accessToken = typeof this.accessToken === 'function'
                ? this.accessToken()
                : this.accessToken;
            requestOptions.headers["Authorization"] = "Bearer " + accessToken;
        }
    }
}
exports.HttpBearerAuth = HttpBearerAuth;
class ApiKeyAuth {
    constructor(location, paramName) {
        this.location = location;
        this.paramName = paramName;
        this.apiKey = '';
    }
    applyToRequest(requestOptions) {
        if (this.location == "query") {
            requestOptions.qs[this.paramName] = this.apiKey;
        }
        else if (this.location == "header" && requestOptions && requestOptions.headers) {
            requestOptions.headers[this.paramName] = this.apiKey;
        }
        else if (this.location == 'cookie' && requestOptions && requestOptions.headers) {
            if (requestOptions.headers['Cookie']) {
                requestOptions.headers['Cookie'] += '; ' + this.paramName + '=' + encodeURIComponent(this.apiKey);
            }
            else {
                requestOptions.headers['Cookie'] = this.paramName + '=' + encodeURIComponent(this.apiKey);
            }
        }
    }
}
exports.ApiKeyAuth = ApiKeyAuth;
class OAuth {
    constructor() {
        this.accessToken = '';
    }
    applyToRequest(requestOptions) {
        if (requestOptions && requestOptions.headers) {
            requestOptions.headers["Authorization"] = "Bearer " + this.accessToken;
        }
    }
}
exports.OAuth = OAuth;
class VoidAuth {
    constructor() {
        this.username = '';
        this.password = '';
    }
    applyToRequest(_) {
    }
}
exports.VoidAuth = VoidAuth;
