// testit-api-client/dist/model/webHookEventType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebHookEventType = void 0;
var WebHookEventType;
(function (WebHookEventType) {
    WebHookEventType[WebHookEventType["AutomatedTestRunCreated"] = 'AutomatedTestRunCreated'] = "AutomatedTestRunCreated";
    WebHookEventType[WebHookEventType["TestPlansStatusChanged"] = 'TestPlansStatusChanged'] = "TestPlansStatusChanged";
    WebHookEventType[WebHookEventType["TestRunStopped"] = 'TestRunStopped'] = "TestRunStopped";
    WebHookEventType[WebHookEventType["TestPointAssigned"] = 'TestPointAssigned'] = "TestPointAssigned";
    WebHookEventType[WebHookEventType["TestResultJiraIssueCreated"] = 'TestResultJiraIssueCreated'] = "TestResultJiraIssueCreated";
    WebHookEventType[WebHookEventType["AutoTestFinished"] = 'AutoTestFinished'] = "AutoTestFinished";
    WebHookEventType[WebHookEventType["UserMentionedInComment"] = 'UserMentionedInComment'] = "UserMentionedInComment";
    WebHookEventType[WebHookEventType["UserSelectedInWorkItemAttribute"] = 'UserSelectedInWorkItemAttribute'] = "UserSelectedInWorkItemAttribute";
    WebHookEventType[WebHookEventType["AllTestPointsFinished"] = 'AllTestPointsFinished'] = "AllTestPointsFinished";
    WebHookEventType[WebHookEventType["AllAutoTestsFinished"] = 'AllAutoTestsFinished'] = "AllAutoTestsFinished";
    WebHookEventType[WebHookEventType["AutoTestChanged"] = 'AutoTestChanged'] = "AutoTestChanged";
    WebHookEventType[WebHookEventType["WorkItemAutoTestRelationChanged"] = 'WorkItemAutoTestRelationChanged'] = "WorkItemAutoTestRelationChanged";
    WebHookEventType[WebHookEventType["WorkItemAttributeChanged"] = 'WorkItemAttributeChanged'] = "WorkItemAttributeChanged";
    WebHookEventType[WebHookEventType["WorkItemChanged"] = 'WorkItemChanged'] = "WorkItemChanged";
    WebHookEventType[WebHookEventType["ConfigurationChanged"] = 'ConfigurationChanged'] = "ConfigurationChanged";
    WebHookEventType[WebHookEventType["ProjectChanged"] = 'ProjectChanged'] = "ProjectChanged";
    WebHookEventType[WebHookEventType["TestPlanChanged"] = 'TestPlanChanged'] = "TestPlanChanged";
})(WebHookEventType = exports.WebHookEventType || (exports.WebHookEventType = {}));
