// testit-api-client/dist/api/apis.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.APIS = exports.HttpError = void 0;
__exportStar(require("./attachmentsApi"), exports);
const attachmentsApi_1 = require("./attachmentsApi");
__exportStar(require("./autoTestsApi"), exports);
const autoTestsApi_1 = require("./autoTestsApi");
__exportStar(require("./backgroundJobsApi"), exports);
const backgroundJobsApi_1 = require("./backgroundJobsApi");
__exportStar(require("./configurationsApi"), exports);
const configurationsApi_1 = require("./configurationsApi");
__exportStar(require("./customAttributeTemplatesApi"), exports);
const customAttributeTemplatesApi_1 = require("./customAttributeTemplatesApi");
__exportStar(require("./customAttributesApi"), exports);
const customAttributesApi_1 = require("./customAttributesApi");
__exportStar(require("./notificationsApi"), exports);
const notificationsApi_1 = require("./notificationsApi");
__exportStar(require("./parametersApi"), exports);
const parametersApi_1 = require("./parametersApi");
__exportStar(require("./projectAttributeTemplatesApi"), exports);
const projectAttributeTemplatesApi_1 = require("./projectAttributeTemplatesApi");
__exportStar(require("./projectAttributesApi"), exports);
const projectAttributesApi_1 = require("./projectAttributesApi");
__exportStar(require("./projectConfigurationsApi"), exports);
const projectConfigurationsApi_1 = require("./projectConfigurationsApi");
__exportStar(require("./projectExportApi"), exports);
const projectExportApi_1 = require("./projectExportApi");
__exportStar(require("./projectImportApi"), exports);
const projectImportApi_1 = require("./projectImportApi");
__exportStar(require("./projectSectionsApi"), exports);
const projectSectionsApi_1 = require("./projectSectionsApi");
__exportStar(require("./projectTestPlanAttributesApi"), exports);
const projectTestPlanAttributesApi_1 = require("./projectTestPlanAttributesApi");
__exportStar(require("./projectTestPlansApi"), exports);
const projectTestPlansApi_1 = require("./projectTestPlansApi");
__exportStar(require("./projectWorkItemsApi"), exports);
const projectWorkItemsApi_1 = require("./projectWorkItemsApi");
__exportStar(require("./projectsApi"), exports);
const projectsApi_1 = require("./projectsApi");
__exportStar(require("./searchApi"), exports);
const searchApi_1 = require("./searchApi");
__exportStar(require("./sectionsApi"), exports);
const sectionsApi_1 = require("./sectionsApi");
__exportStar(require("./tagsApi"), exports);
const tagsApi_1 = require("./tagsApi");
__exportStar(require("./testPlansApi"), exports);
const testPlansApi_1 = require("./testPlansApi");
__exportStar(require("./testPointsApi"), exports);
const testPointsApi_1 = require("./testPointsApi");
__exportStar(require("./testResultsApi"), exports);
const testResultsApi_1 = require("./testResultsApi");
__exportStar(require("./testRunsApi"), exports);
const testRunsApi_1 = require("./testRunsApi");
__exportStar(require("./testSuitesApi"), exports);
const testSuitesApi_1 = require("./testSuitesApi");
__exportStar(require("./webhooksApi"), exports);
const webhooksApi_1 = require("./webhooksApi");
__exportStar(require("./webhooksLogsApi"), exports);
const webhooksLogsApi_1 = require("./webhooksLogsApi");
__exportStar(require("./workItemsApi"), exports);
const workItemsApi_1 = require("./workItemsApi");
__exportStar(require("./workItemsCommentsApi"), exports);
const workItemsCommentsApi_1 = require("./workItemsCommentsApi");
class HttpError extends Error {
    constructor(response, body, statusCode) {
        super('HTTP request failed');
        this.response = response;
        this.body = body;
        this.statusCode = statusCode;
        this.name = 'HttpError';
    }
}
exports.HttpError = HttpError;
exports.APIS = [attachmentsApi_1.AttachmentsApi, autoTestsApi_1.AutoTestsApi, backgroundJobsApi_1.BackgroundJobsApi, configurationsApi_1.ConfigurationsApi, customAttributeTemplatesApi_1.CustomAttributeTemplatesApi, customAttributesApi_1.CustomAttributesApi, notificationsApi_1.NotificationsApi, parametersApi_1.ParametersApi, projectAttributeTemplatesApi_1.ProjectAttributeTemplatesApi, projectAttributesApi_1.ProjectAttributesApi, projectConfigurationsApi_1.ProjectConfigurationsApi, projectExportApi_1.ProjectExportApi, projectImportApi_1.ProjectImportApi, projectSectionsApi_1.ProjectSectionsApi, projectTestPlanAttributesApi_1.ProjectTestPlanAttributesApi, projectTestPlansApi_1.ProjectTestPlansApi, projectWorkItemsApi_1.ProjectWorkItemsApi, projectsApi_1.ProjectsApi, searchApi_1.SearchApi, sectionsApi_1.SectionsApi, tagsApi_1.TagsApi, testPlansApi_1.TestPlansApi, testPointsApi_1.TestPointsApi, testResultsApi_1.TestResultsApi, testRunsApi_1.TestRunsApi, testSuitesApi_1.TestSuitesApi, webhooksApi_1.WebhooksApi, webhooksLogsApi_1.WebhooksLogsApi, workItemsApi_1.WorkItemsApi, workItemsCommentsApi_1.WorkItemsCommentsApi];
