// testit-api-client/dist/api/projectWorkItemsApi.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProjectWorkItemsApi = exports.ProjectWorkItemsApiApiKeys = void 0;
const request_1 = __importDefault(require("request"));
const models_1 = require("../model/models");
const models_2 = require("../model/models");
const apis_1 = require("./apis");
let defaultBasePath = 'http://localhost';
var ProjectWorkItemsApiApiKeys;
(function (ProjectWorkItemsApiApiKeys) {
    ProjectWorkItemsApiApiKeys[ProjectWorkItemsApiApiKeys["Bearer or PrivateToken"] = 0] = "Bearer or PrivateToken";
})(ProjectWorkItemsApiApiKeys = exports.ProjectWorkItemsApiApiKeys || (exports.ProjectWorkItemsApiApiKeys = {}));
class ProjectWorkItemsApi {
    constructor(basePathOrUsername, password, basePath) {
        this._basePath = defaultBasePath;
        this._defaultHeaders = {};
        this._useQuerystring = false;
        this.authentications = {
            'default': new models_1.VoidAuth(),
            'Bearer or PrivateToken': new models_2.ApiKeyAuth('header', 'Authorization'),
        };
        this.interceptors = [];
        if (password) {
            if (basePath) {
                this.basePath = basePath;
            }
        }
        else {
            if (basePathOrUsername) {
                this.basePath = basePathOrUsername;
            }
        }
    }
    set useQuerystring(value) {
        this._useQuerystring = value;
    }
    set basePath(basePath) {
        this._basePath = basePath;
    }
    set defaultHeaders(defaultHeaders) {
        this._defaultHeaders = defaultHeaders;
    }
    get defaultHeaders() {
        return this._defaultHeaders;
    }
    get basePath() {
        return this._basePath;
    }
    setDefaultAuthentication(auth) {
        this.authentications.default = auth;
    }
    setApiKey(key, value) {
        this.authentications[ProjectWorkItemsApiApiKeys[key]].apiKey = value;
    }
    addInterceptor(interceptor) {
        this.interceptors.push(interceptor);
    }
    apiV2ProjectsProjectIdWorkItemsSearchGroupedPost(projectId, skip, take, orderBy, searchField, searchValue, workItemGroupGetModel, options = { headers: {} }) {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.basePath + '/api/v2/projects/{projectId}/workItems/search/grouped'
                .replace('{' + 'projectId' + '}', encodeURIComponent(String(projectId)));
            let localVarQueryParameters = {};
            let localVarHeaderParams = Object.assign({}, this._defaultHeaders);
            const produces = ['application/json'];
            if (produces.indexOf('application/json') >= 0) {
                localVarHeaderParams.Accept = 'application/json';
            }
            else {
                localVarHeaderParams.Accept = produces.join(',');
            }
            let localVarFormParams = {};
            if (projectId === null || projectId === undefined) {
                throw new Error('Required parameter projectId was null or undefined when calling apiV2ProjectsProjectIdWorkItemsSearchGroupedPost.');
            }
            if (skip !== undefined) {
                localVarQueryParameters['Skip'] = models_1.ObjectSerializer.serialize(skip, "number");
            }
            if (take !== undefined) {
                localVarQueryParameters['Take'] = models_1.ObjectSerializer.serialize(take, "number");
            }
            if (orderBy !== undefined) {
                localVarQueryParameters['OrderBy'] = models_1.ObjectSerializer.serialize(orderBy, "string");
            }
            if (searchField !== undefined) {
                localVarQueryParameters['SearchField'] = models_1.ObjectSerializer.serialize(searchField, "string");
            }
            if (searchValue !== undefined) {
                localVarQueryParameters['SearchValue'] = models_1.ObjectSerializer.serialize(searchValue, "string");
            }
            Object.assign(localVarHeaderParams, options.headers);
            let localVarUseFormData = false;
            let localVarRequestOptions = {
                method: 'POST',
                qs: localVarQueryParameters,
                headers: localVarHeaderParams,
                uri: localVarPath,
                useQuerystring: this._useQuerystring,
                json: true,
                body: models_1.ObjectSerializer.serialize(workItemGroupGetModel, "WorkItemGroupGetModel")
            };
            let authenticationPromise = Promise.resolve();
            if (this.authentications['Bearer or PrivateToken'].apiKey) {
                authenticationPromise = authenticationPromise.then(() => this.authentications['Bearer or PrivateToken'].applyToRequest(localVarRequestOptions));
            }
            authenticationPromise = authenticationPromise.then(() => this.authentications.default.applyToRequest(localVarRequestOptions));
            let interceptorPromise = authenticationPromise;
            for (const interceptor of this.interceptors) {
                interceptorPromise = interceptorPromise.then(() => interceptor(localVarRequestOptions));
            }
            return interceptorPromise.then(() => {
                if (Object.keys(localVarFormParams).length) {
                    if (localVarUseFormData) {
                        localVarRequestOptions.formData = localVarFormParams;
                    }
                    else {
                        localVarRequestOptions.form = localVarFormParams;
                    }
                }
                return new Promise((resolve, reject) => {
                    (0, request_1.default)(localVarRequestOptions, (error, response, body) => {
                        if (error) {
                            reject(error);
                        }
                        else {
                            if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
                                body = models_1.ObjectSerializer.deserialize(body, "Array<WorkItemGroupModel>");
                                resolve({ response: response, body: body });
                            }
                            else {
                                reject(new apis_1.HttpError(response, body, response.statusCode));
                            }
                        }
                    });
                });
            });
        });
    }
    apiV2ProjectsProjectIdWorkItemsSearchIdPost(projectId, skip, take, orderBy, searchField, searchValue, workItemSelectModel, options = { headers: {} }) {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.basePath + '/api/v2/projects/{projectId}/workItems/search/id'
                .replace('{' + 'projectId' + '}', encodeURIComponent(String(projectId)));
            let localVarQueryParameters = {};
            let localVarHeaderParams = Object.assign({}, this._defaultHeaders);
            const produces = ['application/json'];
            if (produces.indexOf('application/json') >= 0) {
                localVarHeaderParams.Accept = 'application/json';
            }
            else {
                localVarHeaderParams.Accept = produces.join(',');
            }
            let localVarFormParams = {};
            if (projectId === null || projectId === undefined) {
                throw new Error('Required parameter projectId was null or undefined when calling apiV2ProjectsProjectIdWorkItemsSearchIdPost.');
            }
            if (skip !== undefined) {
                localVarQueryParameters['Skip'] = models_1.ObjectSerializer.serialize(skip, "number");
            }
            if (take !== undefined) {
                localVarQueryParameters['Take'] = models_1.ObjectSerializer.serialize(take, "number");
            }
            if (orderBy !== undefined) {
                localVarQueryParameters['OrderBy'] = models_1.ObjectSerializer.serialize(orderBy, "string");
            }
            if (searchField !== undefined) {
                localVarQueryParameters['SearchField'] = models_1.ObjectSerializer.serialize(searchField, "string");
            }
            if (searchValue !== undefined) {
                localVarQueryParameters['SearchValue'] = models_1.ObjectSerializer.serialize(searchValue, "string");
            }
            Object.assign(localVarHeaderParams, options.headers);
            let localVarUseFormData = false;
            let localVarRequestOptions = {
                method: 'POST',
                qs: localVarQueryParameters,
                headers: localVarHeaderParams,
                uri: localVarPath,
                useQuerystring: this._useQuerystring,
                json: true,
                body: models_1.ObjectSerializer.serialize(workItemSelectModel, "WorkItemSelectModel")
            };
            let authenticationPromise = Promise.resolve();
            if (this.authentications['Bearer or PrivateToken'].apiKey) {
                authenticationPromise = authenticationPromise.then(() => this.authentications['Bearer or PrivateToken'].applyToRequest(localVarRequestOptions));
            }
            authenticationPromise = authenticationPromise.then(() => this.authentications.default.applyToRequest(localVarRequestOptions));
            let interceptorPromise = authenticationPromise;
            for (const interceptor of this.interceptors) {
                interceptorPromise = interceptorPromise.then(() => interceptor(localVarRequestOptions));
            }
            return interceptorPromise.then(() => {
                if (Object.keys(localVarFormParams).length) {
                    if (localVarUseFormData) {
                        localVarRequestOptions.formData = localVarFormParams;
                    }
                    else {
                        localVarRequestOptions.form = localVarFormParams;
                    }
                }
                return new Promise((resolve, reject) => {
                    (0, request_1.default)(localVarRequestOptions, (error, response, body) => {
                        if (error) {
                            reject(error);
                        }
                        else {
                            if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
                                body = models_1.ObjectSerializer.deserialize(body, "Array<string>");
                                resolve({ response: response, body: body });
                            }
                            else {
                                reject(new apis_1.HttpError(response, body, response.statusCode));
                            }
                        }
                    });
                });
            });
        });
    }
    apiV2ProjectsProjectIdWorkItemsSearchPost(projectId, skip, take, orderBy, searchField, searchValue, workItemSelectModel, options = { headers: {} }) {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.basePath + '/api/v2/projects/{projectId}/workItems/search'
                .replace('{' + 'projectId' + '}', encodeURIComponent(String(projectId)));
            let localVarQueryParameters = {};
            let localVarHeaderParams = Object.assign({}, this._defaultHeaders);
            const produces = ['application/json'];
            if (produces.indexOf('application/json') >= 0) {
                localVarHeaderParams.Accept = 'application/json';
            }
            else {
                localVarHeaderParams.Accept = produces.join(',');
            }
            let localVarFormParams = {};
            if (projectId === null || projectId === undefined) {
                throw new Error('Required parameter projectId was null or undefined when calling apiV2ProjectsProjectIdWorkItemsSearchPost.');
            }
            if (skip !== undefined) {
                localVarQueryParameters['Skip'] = models_1.ObjectSerializer.serialize(skip, "number");
            }
            if (take !== undefined) {
                localVarQueryParameters['Take'] = models_1.ObjectSerializer.serialize(take, "number");
            }
            if (orderBy !== undefined) {
                localVarQueryParameters['OrderBy'] = models_1.ObjectSerializer.serialize(orderBy, "string");
            }
            if (searchField !== undefined) {
                localVarQueryParameters['SearchField'] = models_1.ObjectSerializer.serialize(searchField, "string");
            }
            if (searchValue !== undefined) {
                localVarQueryParameters['SearchValue'] = models_1.ObjectSerializer.serialize(searchValue, "string");
            }
            Object.assign(localVarHeaderParams, options.headers);
            let localVarUseFormData = false;
            let localVarRequestOptions = {
                method: 'POST',
                qs: localVarQueryParameters,
                headers: localVarHeaderParams,
                uri: localVarPath,
                useQuerystring: this._useQuerystring,
                json: true,
                body: models_1.ObjectSerializer.serialize(workItemSelectModel, "WorkItemSelectModel")
            };
            let authenticationPromise = Promise.resolve();
            if (this.authentications['Bearer or PrivateToken'].apiKey) {
                authenticationPromise = authenticationPromise.then(() => this.authentications['Bearer or PrivateToken'].applyToRequest(localVarRequestOptions));
            }
            authenticationPromise = authenticationPromise.then(() => this.authentications.default.applyToRequest(localVarRequestOptions));
            let interceptorPromise = authenticationPromise;
            for (const interceptor of this.interceptors) {
                interceptorPromise = interceptorPromise.then(() => interceptor(localVarRequestOptions));
            }
            return interceptorPromise.then(() => {
                if (Object.keys(localVarFormParams).length) {
                    if (localVarUseFormData) {
                        localVarRequestOptions.formData = localVarFormParams;
                    }
                    else {
                        localVarRequestOptions.form = localVarFormParams;
                    }
                }
                return new Promise((resolve, reject) => {
                    (0, request_1.default)(localVarRequestOptions, (error, response, body) => {
                        if (error) {
                            reject(error);
                        }
                        else {
                            if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
                                body = models_1.ObjectSerializer.deserialize(body, "Array<WorkItemShortModel>");
                                resolve({ response: response, body: body });
                            }
                            else {
                                reject(new apis_1.HttpError(response, body, response.statusCode));
                            }
                        }
                    });
                });
            });
        });
    }
    apiV2ProjectsProjectIdWorkItemsTagsGet(projectId, isDeleted, options = { headers: {} }) {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.basePath + '/api/v2/projects/{projectId}/workItems/tags'
                .replace('{' + 'projectId' + '}', encodeURIComponent(String(projectId)));
            let localVarQueryParameters = {};
            let localVarHeaderParams = Object.assign({}, this._defaultHeaders);
            const produces = ['application/json'];
            if (produces.indexOf('application/json') >= 0) {
                localVarHeaderParams.Accept = 'application/json';
            }
            else {
                localVarHeaderParams.Accept = produces.join(',');
            }
            let localVarFormParams = {};
            if (projectId === null || projectId === undefined) {
                throw new Error('Required parameter projectId was null or undefined when calling apiV2ProjectsProjectIdWorkItemsTagsGet.');
            }
            if (isDeleted !== undefined) {
                localVarQueryParameters['isDeleted'] = models_1.ObjectSerializer.serialize(isDeleted, "boolean");
            }
            Object.assign(localVarHeaderParams, options.headers);
            let localVarUseFormData = false;
            let localVarRequestOptions = {
                method: 'GET',
                qs: localVarQueryParameters,
                headers: localVarHeaderParams,
                uri: localVarPath,
                useQuerystring: this._useQuerystring,
                json: true,
            };
            let authenticationPromise = Promise.resolve();
            if (this.authentications['Bearer or PrivateToken'].apiKey) {
                authenticationPromise = authenticationPromise.then(() => this.authentications['Bearer or PrivateToken'].applyToRequest(localVarRequestOptions));
            }
            authenticationPromise = authenticationPromise.then(() => this.authentications.default.applyToRequest(localVarRequestOptions));
            let interceptorPromise = authenticationPromise;
            for (const interceptor of this.interceptors) {
                interceptorPromise = interceptorPromise.then(() => interceptor(localVarRequestOptions));
            }
            return interceptorPromise.then(() => {
                if (Object.keys(localVarFormParams).length) {
                    if (localVarUseFormData) {
                        localVarRequestOptions.formData = localVarFormParams;
                    }
                    else {
                        localVarRequestOptions.form = localVarFormParams;
                    }
                }
                return new Promise((resolve, reject) => {
                    (0, request_1.default)(localVarRequestOptions, (error, response, body) => {
                        if (error) {
                            reject(error);
                        }
                        else {
                            if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
                                body = models_1.ObjectSerializer.deserialize(body, "Array<TagShortModel>");
                                resolve({ response: response, body: body });
                            }
                            else {
                                reject(new apis_1.HttpError(response, body, response.statusCode));
                            }
                        }
                    });
                });
            });
        });
    }
    getWorkItemsByProjectId(projectId, isDeleted, tagNames, includeIterations, skip, take, orderBy, searchField, searchValue, options = { headers: {} }) {
        return __awaiter(this, void 0, void 0, function* () {
            const localVarPath = this.basePath + '/api/v2/projects/{projectId}/workItems'
                .replace('{' + 'projectId' + '}', encodeURIComponent(String(projectId)));
            let localVarQueryParameters = {};
            let localVarHeaderParams = Object.assign({}, this._defaultHeaders);
            const produces = ['application/json'];
            if (produces.indexOf('application/json') >= 0) {
                localVarHeaderParams.Accept = 'application/json';
            }
            else {
                localVarHeaderParams.Accept = produces.join(',');
            }
            let localVarFormParams = {};
            if (projectId === null || projectId === undefined) {
                throw new Error('Required parameter projectId was null or undefined when calling getWorkItemsByProjectId.');
            }
            if (isDeleted !== undefined) {
                localVarQueryParameters['isDeleted'] = models_1.ObjectSerializer.serialize(isDeleted, "boolean");
            }
            if (tagNames !== undefined) {
                localVarQueryParameters['tagNames'] = models_1.ObjectSerializer.serialize(tagNames, "Array<string>");
            }
            if (includeIterations !== undefined) {
                localVarQueryParameters['includeIterations'] = models_1.ObjectSerializer.serialize(includeIterations, "boolean");
            }
            if (skip !== undefined) {
                localVarQueryParameters['Skip'] = models_1.ObjectSerializer.serialize(skip, "number");
            }
            if (take !== undefined) {
                localVarQueryParameters['Take'] = models_1.ObjectSerializer.serialize(take, "number");
            }
            if (orderBy !== undefined) {
                localVarQueryParameters['OrderBy'] = models_1.ObjectSerializer.serialize(orderBy, "string");
            }
            if (searchField !== undefined) {
                localVarQueryParameters['SearchField'] = models_1.ObjectSerializer.serialize(searchField, "string");
            }
            if (searchValue !== undefined) {
                localVarQueryParameters['SearchValue'] = models_1.ObjectSerializer.serialize(searchValue, "string");
            }
            Object.assign(localVarHeaderParams, options.headers);
            let localVarUseFormData = false;
            let localVarRequestOptions = {
                method: 'GET',
                qs: localVarQueryParameters,
                headers: localVarHeaderParams,
                uri: localVarPath,
                useQuerystring: this._useQuerystring,
                json: true,
            };
            let authenticationPromise = Promise.resolve();
            if (this.authentications['Bearer or PrivateToken'].apiKey) {
                authenticationPromise = authenticationPromise.then(() => this.authentications['Bearer or PrivateToken'].applyToRequest(localVarRequestOptions));
            }
            authenticationPromise = authenticationPromise.then(() => this.authentications.default.applyToRequest(localVarRequestOptions));
            let interceptorPromise = authenticationPromise;
            for (const interceptor of this.interceptors) {
                interceptorPromise = interceptorPromise.then(() => interceptor(localVarRequestOptions));
            }
            return interceptorPromise.then(() => {
                if (Object.keys(localVarFormParams).length) {
                    if (localVarUseFormData) {
                        localVarRequestOptions.formData = localVarFormParams;
                    }
                    else {
                        localVarRequestOptions.form = localVarFormParams;
                    }
                }
                return new Promise((resolve, reject) => {
                    (0, request_1.default)(localVarRequestOptions, (error, response, body) => {
                        if (error) {
                            reject(error);
                        }
                        else {
                            if (response.statusCode && response.statusCode >= 200 && response.statusCode <= 299) {
                                body = models_1.ObjectSerializer.deserialize(body, "Array<WorkItemShortModel>");
                                resolve({ response: response, body: body });
                            }
                            else {
                                reject(new apis_1.HttpError(response, body, response.statusCode));
                            }
                        }
                    });
                });
            });
        });
    }
}
exports.ProjectWorkItemsApi = ProjectWorkItemsApi;
